// @huaweicloud/huaweicloud-sdk-ecs/v2/model/BatchResetServersPasswordRequestBody.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchResetServersPasswordRequestBody = void 0;
var BatchResetServersPasswordRequestBody = /** @class */ (function () {
    function BatchResetServersPasswordRequestBody(newPassword, servers) {
        this['new_password'] = newPassword;
        this['servers'] = servers;
    }
    BatchResetServersPasswordRequestBody.prototype.withNewPassword = function (newPassword) {
        this['new_password'] = newPassword;
        return this;
    };
    Object.defineProperty(BatchResetServersPasswordRequestBody.prototype, "newPassword", {
        get: function () {
            return this['new_password'];
        },
        set: function (newPassword) {
            this['new_password'] = newPassword;
        },
        enumerable: false,
        configurable: true
    });
    BatchResetServersPasswordRequestBody.prototype.withDryRun = function (dryRun) {
        this['dry_run'] = dryRun;
        return this;
    };
    Object.defineProperty(BatchResetServersPasswordRequestBody.prototype, "dryRun", {
        get: function () {
            return this['dry_run'];
        },
        set: function (dryRun) {
            this['dry_run'] = dryRun;
        },
        enumerable: false,
        configurable: true
    });
    BatchResetServersPasswordRequestBody.prototype.withServers = function (servers) {
        this['servers'] = servers;
        return this;
    };
    return BatchResetServersPasswordRequestBody;
}());
exports.BatchResetServersPasswordRequestBody = BatchResetServersPasswordRequestBody;
