// @huaweicloud/huaweicloud-sdk-ecs/v2/model/BatchCreateServerTagsRequestBody.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchCreateServerTagsRequestBodyActionEnum = exports.BatchCreateServerTagsRequestBody = void 0;
var BatchCreateServerTagsRequestBody = /** @class */ (function () {
    function BatchCreateServerTagsRequestBody(action, tags) {
        this['action'] = action;
        this['tags'] = tags;
    }
    BatchCreateServerTagsRequestBody.prototype.withAction = function (action) {
        this['action'] = action;
        return this;
    };
    BatchCreateServerTagsRequestBody.prototype.withTags = function (tags) {
        this['tags'] = tags;
        return this;
    };
    return BatchCreateServerTagsRequestBody;
}());
exports.BatchCreateServerTagsRequestBody = BatchCreateServerTagsRequestBody;
/**
    * @export
    * @enum {string}
    */
var BatchCreateServerTagsRequestBodyActionEnum;
(function (BatchCreateServerTagsRequestBodyActionEnum) {
    BatchCreateServerTagsRequestBodyActionEnum["CREATE"] = "create";
})(BatchCreateServerTagsRequestBodyActionEnum = exports.BatchCreateServerTagsRequestBodyActionEnum || (exports.BatchCreateServerTagsRequestBodyActionEnum = {}));
