// @huaweicloud/huaweicloud-sdk-ecs/v2/model/BatchUpdateServersNameRequestBody.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchUpdateServersNameRequestBody = void 0;
var BatchUpdateServersNameRequestBody = /** @class */ (function () {
    function BatchUpdateServersNameRequestBody(name, servers) {
        this['name'] = name;
        this['servers'] = servers;
    }
    BatchUpdateServersNameRequestBody.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    BatchUpdateServersNameRequestBody.prototype.withDryRun = function (dryRun) {
        this['dry_run'] = dryRun;
        return this;
    };
    Object.defineProperty(BatchUpdateServersNameRequestBody.prototype, "dryRun", {
        get: function () {
            return this['dry_run'];
        },
        set: function (dryRun) {
            this['dry_run'] = dryRun;
        },
        enumerable: false,
        configurable: true
    });
    BatchUpdateServersNameRequestBody.prototype.withServers = function (servers) {
        this['servers'] = servers;
        return this;
    };
    return BatchUpdateServersNameRequestBody;
}());
exports.BatchUpdateServersNameRequestBody = BatchUpdateServersNameRequestBody;
