// @huaweicloud/huaweicloud-sdk-ecs/v2/model/BatchRebootSeversOption.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchRebootSeversOptionTypeEnum = exports.BatchRebootSeversOption = void 0;
var BatchRebootSeversOption = /** @class */ (function () {
    function BatchRebootSeversOption(servers, type) {
        this['servers'] = servers;
        this['type'] = type;
    }
    BatchRebootSeversOption.prototype.withServers = function (servers) {
        this['servers'] = servers;
        return this;
    };
    BatchRebootSeversOption.prototype.withType = function (type) {
        this['type'] = type;
        return this;
    };
    return BatchRebootSeversOption;
}());
exports.BatchRebootSeversOption = BatchRebootSeversOption;
/**
    * @export
    * @enum {string}
    */
var BatchRebootSeversOptionTypeEnum;
(function (BatchRebootSeversOptionTypeEnum) {
    BatchRebootSeversOptionTypeEnum["SOFT"] = "SOFT";
    BatchRebootSeversOptionTypeEnum["HARD"] = "HARD";
})(BatchRebootSeversOptionTypeEnum = exports.BatchRebootSeversOptionTypeEnum || (exports.BatchRebootSeversOptionTypeEnum = {}));
