// @huaweicloud/huaweicloud-sdk-ecs/v2/model/BatchDeleteServerTagsRequestBody.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchDeleteServerTagsRequestBodyActionEnum = exports.BatchDeleteServerTagsRequestBody = void 0;
var BatchDeleteServerTagsRequestBody = /** @class */ (function () {
    function BatchDeleteServerTagsRequestBody(action, tags) {
        this['action'] = action;
        this['tags'] = tags;
    }
    BatchDeleteServerTagsRequestBody.prototype.withAction = function (action) {
        this['action'] = action;
        return this;
    };
    BatchDeleteServerTagsRequestBody.prototype.withTags = function (tags) {
        this['tags'] = tags;
        return this;
    };
    return BatchDeleteServerTagsRequestBody;
}());
exports.BatchDeleteServerTagsRequestBody = BatchDeleteServerTagsRequestBody;
/**
    * @export
    * @enum {string}
    */
var BatchDeleteServerTagsRequestBodyActionEnum;
(function (BatchDeleteServerTagsRequestBodyActionEnum) {
    BatchDeleteServerTagsRequestBodyActionEnum["DELETE"] = "delete";
})(BatchDeleteServerTagsRequestBodyActionEnum = exports.BatchDeleteServerTagsRequestBodyActionEnum || (exports.BatchDeleteServerTagsRequestBodyActionEnum = {}));
