// @huaweicloud/huaweicloud-sdk-ecs/v2/model/BatchAddServerNicOption.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchAddServerNicOption = void 0;
var BatchAddServerNicOption = /** @class */ (function () {
    function BatchAddServerNicOption(subnetId) {
        this['subnet_id'] = subnetId;
    }
    BatchAddServerNicOption.prototype.withSubnetId = function (subnetId) {
        this['subnet_id'] = subnetId;
        return this;
    };
    Object.defineProperty(BatchAddServerNicOption.prototype, "subnetId", {
        get: function () {
            return this['subnet_id'];
        },
        set: function (subnetId) {
            this['subnet_id'] = subnetId;
        },
        enumerable: false,
        configurable: true
    });
    BatchAddServerNicOption.prototype.withSecurityGroups = function (securityGroups) {
        this['security_groups'] = securityGroups;
        return this;
    };
    Object.defineProperty(BatchAddServerNicOption.prototype, "securityGroups", {
        get: function () {
            return this['security_groups'];
        },
        set: function (securityGroups) {
            this['security_groups'] = securityGroups;
        },
        enumerable: false,
        configurable: true
    });
    BatchAddServerNicOption.prototype.withIpAddress = function (ipAddress) {
        this['ip_address'] = ipAddress;
        return this;
    };
    Object.defineProperty(BatchAddServerNicOption.prototype, "ipAddress", {
        get: function () {
            return this['ip_address'];
        },
        set: function (ipAddress) {
            this['ip_address'] = ipAddress;
        },
        enumerable: false,
        configurable: true
    });
    BatchAddServerNicOption.prototype.withIpv6Enable = function (ipv6Enable) {
        this['ipv6_enable'] = ipv6Enable;
        return this;
    };
    Object.defineProperty(BatchAddServerNicOption.prototype, "ipv6Enable", {
        get: function () {
            return this['ipv6_enable'];
        },
        set: function (ipv6Enable) {
            this['ipv6_enable'] = ipv6Enable;
        },
        enumerable: false,
        configurable: true
    });
    BatchAddServerNicOption.prototype.withIpv6Bandwidth = function (ipv6Bandwidth) {
        this['ipv6_bandwidth'] = ipv6Bandwidth;
        return this;
    };
    Object.defineProperty(BatchAddServerNicOption.prototype, "ipv6Bandwidth", {
        get: function () {
            return this['ipv6_bandwidth'];
        },
        set: function (ipv6Bandwidth) {
            this['ipv6_bandwidth'] = ipv6Bandwidth;
        },
        enumerable: false,
        configurable: true
    });
    return BatchAddServerNicOption;
}());
exports.BatchAddServerNicOption = BatchAddServerNicOption;
