// @huaweicloud/huaweicloud-sdk-ecs/v2/model/BatchStopServersOption.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchStopServersOptionTypeEnum = exports.BatchStopServersOption = void 0;
var BatchStopServersOption = /** @class */ (function () {
    function BatchStopServersOption(servers) {
        this['servers'] = servers;
    }
    BatchStopServersOption.prototype.withServers = function (servers) {
        this['servers'] = servers;
        return this;
    };
    BatchStopServersOption.prototype.withType = function (type) {
        this['type'] = type;
        return this;
    };
    return BatchStopServersOption;
}());
exports.BatchStopServersOption = BatchStopServersOption;
/**
    * @export
    * @enum {string}
    */
var BatchStopServersOptionTypeEnum;
(function (BatchStopServersOptionTypeEnum) {
    BatchStopServersOptionTypeEnum["SOFT"] = "SOFT";
    BatchStopServersOptionTypeEnum["HARD"] = "HARD";
})(BatchStopServersOptionTypeEnum = exports.BatchStopServersOptionTypeEnum || (exports.BatchStopServersOptionTypeEnum = {}));
