// @huaweicloud/huaweicloud-sdk-ecs/v2/model/BlockDeviceAttachableQuantity.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BlockDeviceAttachableQuantity = void 0;
var BlockDeviceAttachableQuantity = /** @class */ (function () {
    function BlockDeviceAttachableQuantity() {
    }
    BlockDeviceAttachableQuantity.prototype.withFreeScsi = function (freeScsi) {
        this['free_scsi'] = freeScsi;
        return this;
    };
    Object.defineProperty(BlockDeviceAttachableQuantity.prototype, "freeScsi", {
        get: function () {
            return this['free_scsi'];
        },
        set: function (freeScsi) {
            this['free_scsi'] = freeScsi;
        },
        enumerable: false,
        configurable: true
    });
    BlockDeviceAttachableQuantity.prototype.withFreeBlk = function (freeBlk) {
        this['free_blk'] = freeBlk;
        return this;
    };
    Object.defineProperty(BlockDeviceAttachableQuantity.prototype, "freeBlk", {
        get: function () {
            return this['free_blk'];
        },
        set: function (freeBlk) {
            this['free_blk'] = freeBlk;
        },
        enumerable: false,
        configurable: true
    });
    BlockDeviceAttachableQuantity.prototype.withFreeDisk = function (freeDisk) {
        this['free_disk'] = freeDisk;
        return this;
    };
    Object.defineProperty(BlockDeviceAttachableQuantity.prototype, "freeDisk", {
        get: function () {
            return this['free_disk'];
        },
        set: function (freeDisk) {
            this['free_disk'] = freeDisk;
        },
        enumerable: false,
        configurable: true
    });
    return BlockDeviceAttachableQuantity;
}());
exports.BlockDeviceAttachableQuantity = BlockDeviceAttachableQuantity;
