// @huaweicloud/huaweicloud-sdk-ecs/v2/model/ChangeServerOsWithCloudInitOption.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChangeServerOsWithCloudInitOption = void 0;
var ChangeServerOsWithCloudInitOption = /** @class */ (function () {
    function ChangeServerOsWithCloudInitOption(imageid) {
        this['imageid'] = imageid;
    }
    ChangeServerOsWithCloudInitOption.prototype.withAdminpass = function (adminpass) {
        this['adminpass'] = adminpass;
        return this;
    };
    ChangeServerOsWithCloudInitOption.prototype.withKeyname = function (keyname) {
        this['keyname'] = keyname;
        return this;
    };
    ChangeServerOsWithCloudInitOption.prototype.withUserid = function (userid) {
        this['userid'] = userid;
        return this;
    };
    ChangeServerOsWithCloudInitOption.prototype.withImageid = function (imageid) {
        this['imageid'] = imageid;
        return this;
    };
    ChangeServerOsWithCloudInitOption.prototype.withMetadata = function (metadata) {
        this['metadata'] = metadata;
        return this;
    };
    ChangeServerOsWithCloudInitOption.prototype.withMode = function (mode) {
        this['mode'] = mode;
        return this;
    };
    return ChangeServerOsWithCloudInitOption;
}());
exports.ChangeServerOsWithCloudInitOption = ChangeServerOsWithCloudInitOption;
