// @huaweicloud/huaweicloud-sdk-ecs/v2/model/ChangeServerOsWithoutCloudInitOption.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChangeServerOsWithoutCloudInitOption = void 0;
var ChangeServerOsWithoutCloudInitOption = /** @class */ (function () {
    function ChangeServerOsWithoutCloudInitOption(imageid) {
        this['imageid'] = imageid;
    }
    ChangeServerOsWithoutCloudInitOption.prototype.withAdminpass = function (adminpass) {
        this['adminpass'] = adminpass;
        return this;
    };
    ChangeServerOsWithoutCloudInitOption.prototype.withKeyname = function (keyname) {
        this['keyname'] = keyname;
        return this;
    };
    ChangeServerOsWithoutCloudInitOption.prototype.withUserid = function (userid) {
        this['userid'] = userid;
        return this;
    };
    ChangeServerOsWithoutCloudInitOption.prototype.withImageid = function (imageid) {
        this['imageid'] = imageid;
        return this;
    };
    ChangeServerOsWithoutCloudInitOption.prototype.withMode = function (mode) {
        this['mode'] = mode;
        return this;
    };
    return ChangeServerOsWithoutCloudInitOption;
}());
exports.ChangeServerOsWithoutCloudInitOption = ChangeServerOsWithoutCloudInitOption;
