// @huaweicloud/huaweicloud-sdk-ecs/v2/model/CreateServerGroupResult.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateServerGroupResult = void 0;
var CreateServerGroupResult = /** @class */ (function () {
    function CreateServerGroupResult(id, members, metadata, name, policies) {
        this['id'] = id;
        this['members'] = members;
        this['metadata'] = metadata;
        this['name'] = name;
        this['policies'] = policies;
    }
    CreateServerGroupResult.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    CreateServerGroupResult.prototype.withMembers = function (members) {
        this['members'] = members;
        return this;
    };
    CreateServerGroupResult.prototype.withMetadata = function (metadata) {
        this['metadata'] = metadata;
        return this;
    };
    CreateServerGroupResult.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    CreateServerGroupResult.prototype.withPolicies = function (policies) {
        this['policies'] = policies;
        return this;
    };
    return CreateServerGroupResult;
}());
exports.CreateServerGroupResult = CreateServerGroupResult;
