// @huaweicloud/huaweicloud-sdk-ecs/v2/model/CreateServerGroupOption.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateServerGroupOptionPoliciesEnum = exports.CreateServerGroupOption = void 0;
var CreateServerGroupOption = /** @class */ (function () {
    function CreateServerGroupOption(name, policies) {
        this['name'] = name;
        this['policies'] = policies;
    }
    CreateServerGroupOption.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    CreateServerGroupOption.prototype.withPolicies = function (policies) {
        this['policies'] = policies;
        return this;
    };
    return CreateServerGroupOption;
}());
exports.CreateServerGroupOption = CreateServerGroupOption;
/**
    * @export
    * @enum {string}
    */
var CreateServerGroupOptionPoliciesEnum;
(function (CreateServerGroupOptionPoliciesEnum) {
    CreateServerGroupOptionPoliciesEnum["ANTI_AFFINITY"] = "anti-affinity";
})(CreateServerGroupOptionPoliciesEnum = exports.CreateServerGroupOptionPoliciesEnum || (exports.CreateServerGroupOptionPoliciesEnum = {}));
