// @huaweicloud/huaweicloud-sdk-ecs/v2/model/DeleteServersRequestBody.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeleteServersRequestBody = void 0;
var DeleteServersRequestBody = /** @class */ (function () {
    function DeleteServersRequestBody(servers) {
        this['servers'] = servers;
    }
    DeleteServersRequestBody.prototype.withDeletePublicip = function (deletePublicip) {
        this['delete_publicip'] = deletePublicip;
        return this;
    };
    Object.defineProperty(DeleteServersRequestBody.prototype, "deletePublicip", {
        get: function () {
            return this['delete_publicip'];
        },
        set: function (deletePublicip) {
            this['delete_publicip'] = deletePublicip;
        },
        enumerable: false,
        configurable: true
    });
    DeleteServersRequestBody.prototype.withDeleteVolume = function (deleteVolume) {
        this['delete_volume'] = deleteVolume;
        return this;
    };
    Object.defineProperty(DeleteServersRequestBody.prototype, "deleteVolume", {
        get: function () {
            return this['delete_volume'];
        },
        set: function (deleteVolume) {
            this['delete_volume'] = deleteVolume;
        },
        enumerable: false,
        configurable: true
    });
    DeleteServersRequestBody.prototype.withServers = function (servers) {
        this['servers'] = servers;
        return this;
    };
    return DeleteServersRequestBody;
}());
exports.DeleteServersRequestBody = DeleteServersRequestBody;
