// @huaweicloud/huaweicloud-sdk-ecs/v2/model/DetachServerVolumeRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DetachServerVolumeRequestDeleteFlagEnum = exports.DetachServerVolumeRequest = void 0;
var DetachServerVolumeRequest = /** @class */ (function () {
    function DetachServerVolumeRequest(serverId, volumeId) {
        this['server_id'] = serverId;
        this['volume_id'] = volumeId;
    }
    DetachServerVolumeRequest.prototype.withServerId = function (serverId) {
        this['server_id'] = serverId;
        return this;
    };
    Object.defineProperty(DetachServerVolumeRequest.prototype, "serverId", {
        get: function () {
            return this['server_id'];
        },
        set: function (serverId) {
            this['server_id'] = serverId;
        },
        enumerable: false,
        configurable: true
    });
    DetachServerVolumeRequest.prototype.withVolumeId = function (volumeId) {
        this['volume_id'] = volumeId;
        return this;
    };
    Object.defineProperty(DetachServerVolumeRequest.prototype, "volumeId", {
        get: function () {
            return this['volume_id'];
        },
        set: function (volumeId) {
            this['volume_id'] = volumeId;
        },
        enumerable: false,
        configurable: true
    });
    DetachServerVolumeRequest.prototype.withDeleteFlag = function (deleteFlag) {
        this['delete_flag'] = deleteFlag;
        return this;
    };
    Object.defineProperty(DetachServerVolumeRequest.prototype, "deleteFlag", {
        get: function () {
            return this['delete_flag'];
        },
        set: function (deleteFlag) {
            this['delete_flag'] = deleteFlag;
        },
        enumerable: false,
        configurable: true
    });
    return DetachServerVolumeRequest;
}());
exports.DetachServerVolumeRequest = DetachServerVolumeRequest;
/**
    * @export
    * @enum {string}
    */
var DetachServerVolumeRequestDeleteFlagEnum;
(function (DetachServerVolumeRequestDeleteFlagEnum) {
    DetachServerVolumeRequestDeleteFlagEnum["E_0"] = "0";
    DetachServerVolumeRequestDeleteFlagEnum["E_1"] = "1";
})(DetachServerVolumeRequestDeleteFlagEnum = exports.DetachServerVolumeRequestDeleteFlagEnum || (exports.DetachServerVolumeRequestDeleteFlagEnum = {}));
