// @huaweicloud/huaweicloud-sdk-ecs/v2/model/DisassociateServerVirtualIpOption.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DisassociateServerVirtualIpOptionIpAddressEnum = exports.DisassociateServerVirtualIpOptionSubnetIdEnum = exports.DisassociateServerVirtualIpOption = void 0;
var DisassociateServerVirtualIpOption = /** @class */ (function () {
    function DisassociateServerVirtualIpOption(subnetId, ipAddress) {
        this['subnet_id'] = subnetId;
        this['ip_address'] = ipAddress;
    }
    DisassociateServerVirtualIpOption.prototype.withSubnetId = function (subnetId) {
        this['subnet_id'] = subnetId;
        return this;
    };
    Object.defineProperty(DisassociateServerVirtualIpOption.prototype, "subnetId", {
        get: function () {
            return this['subnet_id'];
        },
        set: function (subnetId) {
            this['subnet_id'] = subnetId;
        },
        enumerable: false,
        configurable: true
    });
    DisassociateServerVirtualIpOption.prototype.withIpAddress = function (ipAddress) {
        this['ip_address'] = ipAddress;
        return this;
    };
    Object.defineProperty(DisassociateServerVirtualIpOption.prototype, "ipAddress", {
        get: function () {
            return this['ip_address'];
        },
        set: function (ipAddress) {
            this['ip_address'] = ipAddress;
        },
        enumerable: false,
        configurable: true
    });
    DisassociateServerVirtualIpOption.prototype.withReverseBinding = function (reverseBinding) {
        this['reverse_binding'] = reverseBinding;
        return this;
    };
    Object.defineProperty(DisassociateServerVirtualIpOption.prototype, "reverseBinding", {
        get: function () {
            return this['reverse_binding'];
        },
        set: function (reverseBinding) {
            this['reverse_binding'] = reverseBinding;
        },
        enumerable: false,
        configurable: true
    });
    return DisassociateServerVirtualIpOption;
}());
exports.DisassociateServerVirtualIpOption = DisassociateServerVirtualIpOption;
/**
    * @export
    * @enum {string}
    */
var DisassociateServerVirtualIpOptionSubnetIdEnum;
(function (DisassociateServerVirtualIpOptionSubnetIdEnum) {
    DisassociateServerVirtualIpOptionSubnetIdEnum["Empty"] = "";
})(DisassociateServerVirtualIpOptionSubnetIdEnum = exports.DisassociateServerVirtualIpOptionSubnetIdEnum || (exports.DisassociateServerVirtualIpOptionSubnetIdEnum = {}));
/**
    * @export
    * @enum {string}
    */
var DisassociateServerVirtualIpOptionIpAddressEnum;
(function (DisassociateServerVirtualIpOptionIpAddressEnum) {
    DisassociateServerVirtualIpOptionIpAddressEnum["Empty"] = "";
})(DisassociateServerVirtualIpOptionIpAddressEnum = exports.DisassociateServerVirtualIpOptionIpAddressEnum || (exports.DisassociateServerVirtualIpOptionIpAddressEnum = {}));
