// @huaweicloud/huaweicloud-sdk-ecs/v2/model/FlavorExtraSpec.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FlavorExtraSpec = void 0;
var FlavorExtraSpec = /** @class */ (function () {
    function FlavorExtraSpec() {
    }
    FlavorExtraSpec.prototype.withEcsPerformancetype = function (ecsPerformancetype) {
        this['ecs:performancetype'] = ecsPerformancetype;
        return this;
    };
    Object.defineProperty(FlavorExtraSpec.prototype, "ecsPerformancetype", {
        get: function () {
            return this['ecs:performancetype'];
        },
        set: function (ecsPerformancetype) {
            this['ecs:performancetype'] = ecsPerformancetype;
        },
        enumerable: false,
        configurable: true
    });
    FlavorExtraSpec.prototype.withHwNumaNodes = function (hwNumaNodes) {
        this['hw:numa_nodes'] = hwNumaNodes;
        return this;
    };
    Object.defineProperty(FlavorExtraSpec.prototype, "hwNumaNodes", {
        get: function () {
            return this['hw:numa_nodes'];
        },
        set: function (hwNumaNodes) {
            this['hw:numa_nodes'] = hwNumaNodes;
        },
        enumerable: false,
        configurable: true
    });
    FlavorExtraSpec.prototype.withResourceType = function (resourceType) {
        this['resource_type'] = resourceType;
        return this;
    };
    Object.defineProperty(FlavorExtraSpec.prototype, "resourceType", {
        get: function () {
            return this['resource_type'];
        },
        set: function (resourceType) {
            this['resource_type'] = resourceType;
        },
        enumerable: false,
        configurable: true
    });
    FlavorExtraSpec.prototype.withHpetSupport = function (hpetSupport) {
        this['hpet_support'] = hpetSupport;
        return this;
    };
    Object.defineProperty(FlavorExtraSpec.prototype, "hpetSupport", {
        get: function () {
            return this['hpet_support'];
        },
        set: function (hpetSupport) {
            this['hpet_support'] = hpetSupport;
        },
        enumerable: false,
        configurable: true
    });
    FlavorExtraSpec.prototype.withInstanceVnicType = function (instanceVnicType) {
        this['instance_vnic:type'] = instanceVnicType;
        return this;
    };
    Object.defineProperty(FlavorExtraSpec.prototype, "instanceVnicType", {
        get: function () {
            return this['instance_vnic:type'];
        },
        set: function (instanceVnicType) {
            this['instance_vnic:type'] = instanceVnicType;
        },
        enumerable: false,
        configurable: true
    });
    FlavorExtraSpec.prototype.withInstanceVnicInstanceBandwidth = function (instanceVnicInstanceBandwidth) {
        this['instance_vnic:instance_bandwidth'] = instanceVnicInstanceBandwidth;
        return this;
    };
    Object.defineProperty(FlavorExtraSpec.prototype, "instanceVnicInstanceBandwidth", {
        get: function () {
            return this['instance_vnic:instance_bandwidth'];
        },
        set: function (instanceVnicInstanceBandwidth) {
            this['instance_vnic:instance_bandwidth'] = instanceVnicInstanceBandwidth;
        },
        enumerable: false,
        configurable: true
    });
    FlavorExtraSpec.prototype.withInstanceVnicMaxCount = function (instanceVnicMaxCount) {
        this['instance_vnic:max_count'] = instanceVnicMaxCount;
        return this;
    };
    Object.defineProperty(FlavorExtraSpec.prototype, "instanceVnicMaxCount", {
        get: function () {
            return this['instance_vnic:max_count'];
        },
        set: function (instanceVnicMaxCount) {
            this['instance_vnic:max_count'] = instanceVnicMaxCount;
        },
        enumerable: false,
        configurable: true
    });
    FlavorExtraSpec.prototype.withQuotaLocalDisk = function (quotaLocalDisk) {
        this['quota:local_disk'] = quotaLocalDisk;
        return this;
    };
    Object.defineProperty(FlavorExtraSpec.prototype, "quotaLocalDisk", {
        get: function () {
            return this['quota:local_disk'];
        },
        set: function (quotaLocalDisk) {
            this['quota:local_disk'] = quotaLocalDisk;
        },
        enumerable: false,
        configurable: true
    });
    FlavorExtraSpec.prototype.withQuotaNvmeSsd = function (quotaNvmeSsd) {
        this['quota:nvme_ssd'] = quotaNvmeSsd;
        return this;
    };
    Object.defineProperty(FlavorExtraSpec.prototype, "quotaNvmeSsd", {
        get: function () {
            return this['quota:nvme_ssd'];
        },
        set: function (quotaNvmeSsd) {
            this['quota:nvme_ssd'] = quotaNvmeSsd;
        },
        enumerable: false,
        configurable: true
    });
    FlavorExtraSpec.prototype.withExtraSpecIoPersistentGrant = function (extraSpecIoPersistentGrant) {
        this['extra_spec:io:persistent_grant'] = extraSpecIoPersistentGrant;
        return this;
    };
    Object.defineProperty(FlavorExtraSpec.prototype, "extraSpecIoPersistentGrant", {
        get: function () {
            return this['extra_spec:io:persistent_grant'];
        },
        set: function (extraSpecIoPersistentGrant) {
            this['extra_spec:io:persistent_grant'] = extraSpecIoPersistentGrant;
        },
        enumerable: false,
        configurable: true
    });
    FlavorExtraSpec.prototype.withEcsGeneration = function (ecsGeneration) {
        this['ecs:generation'] = ecsGeneration;
        return this;
    };
    Object.defineProperty(FlavorExtraSpec.prototype, "ecsGeneration", {
        get: function () {
            return this['ecs:generation'];
        },
        set: function (ecsGeneration) {
            this['ecs:generation'] = ecsGeneration;
        },
        enumerable: false,
        configurable: true
    });
    FlavorExtraSpec.prototype.withEcsVirtualizationEnvTypes = function (ecsVirtualizationEnvTypes) {
        this['ecs:virtualization_env_types'] = ecsVirtualizationEnvTypes;
        return this;
    };
    Object.defineProperty(FlavorExtraSpec.prototype, "ecsVirtualizationEnvTypes", {
        get: function () {
            return this['ecs:virtualization_env_types'];
        },
        set: function (ecsVirtualizationEnvTypes) {
            this['ecs:virtualization_env_types'] = ecsVirtualizationEnvTypes;
        },
        enumerable: false,
        configurable: true
    });
    FlavorExtraSpec.prototype.withPciPassthroughEnableGpu = function (pciPassthroughEnableGpu) {
        this['pci_passthrough:enable_gpu'] = pciPassthroughEnableGpu;
        return this;
    };
    Object.defineProperty(FlavorExtraSpec.prototype, "pciPassthroughEnableGpu", {
        get: function () {
            return this['pci_passthrough:enable_gpu'];
        },
        set: function (pciPassthroughEnableGpu) {
            this['pci_passthrough:enable_gpu'] = pciPassthroughEnableGpu;
        },
        enumerable: false,
        configurable: true
    });
    FlavorExtraSpec.prototype.withPciPassthroughGpuSpecs = function (pciPassthroughGpuSpecs) {
        this['pci_passthrough:gpu_specs'] = pciPassthroughGpuSpecs;
        return this;
    };
    Object.defineProperty(FlavorExtraSpec.prototype, "pciPassthroughGpuSpecs", {
        get: function () {
            return this['pci_passthrough:gpu_specs'];
        },
        set: function (pciPassthroughGpuSpecs) {
            this['pci_passthrough:gpu_specs'] = pciPassthroughGpuSpecs;
        },
        enumerable: false,
        configurable: true
    });
    FlavorExtraSpec.prototype.withPciPassthroughAlias = function (pciPassthroughAlias) {
        this['pci_passthrough:alias'] = pciPassthroughAlias;
        return this;
    };
    Object.defineProperty(FlavorExtraSpec.prototype, "pciPassthroughAlias", {
        get: function () {
            return this['pci_passthrough:alias'];
        },
        set: function (pciPassthroughAlias) {
            this['pci_passthrough:alias'] = pciPassthroughAlias;
        },
        enumerable: false,
        configurable: true
    });
    FlavorExtraSpec.prototype.withCondOperationStatus = function (condOperationStatus) {
        this['cond:operation:status'] = condOperationStatus;
        return this;
    };
    Object.defineProperty(FlavorExtraSpec.prototype, "condOperationStatus", {
        get: function () {
            return this['cond:operation:status'];
        },
        set: function (condOperationStatus) {
            this['cond:operation:status'] = condOperationStatus;
        },
        enumerable: false,
        configurable: true
    });
    FlavorExtraSpec.prototype.withCondOperationAz = function (condOperationAz) {
        this['cond:operation:az'] = condOperationAz;
        return this;
    };
    Object.defineProperty(FlavorExtraSpec.prototype, "condOperationAz", {
        get: function () {
            return this['cond:operation:az'];
        },
        set: function (condOperationAz) {
            this['cond:operation:az'] = condOperationAz;
        },
        enumerable: false,
        configurable: true
    });
    FlavorExtraSpec.prototype.withQuotaMaxRate = function (quotaMaxRate) {
        this['quota:max_rate'] = quotaMaxRate;
        return this;
    };
    Object.defineProperty(FlavorExtraSpec.prototype, "quotaMaxRate", {
        get: function () {
            return this['quota:max_rate'];
        },
        set: function (quotaMaxRate) {
            this['quota:max_rate'] = quotaMaxRate;
        },
        enumerable: false,
        configurable: true
    });
    FlavorExtraSpec.prototype.withQuotaMinRate = function (quotaMinRate) {
        this['quota:min_rate'] = quotaMinRate;
        return this;
    };
    Object.defineProperty(FlavorExtraSpec.prototype, "quotaMinRate", {
        get: function () {
            return this['quota:min_rate'];
        },
        set: function (quotaMinRate) {
            this['quota:min_rate'] = quotaMinRate;
        },
        enumerable: false,
        configurable: true
    });
    FlavorExtraSpec.prototype.withQuotaMaxPps = function (quotaMaxPps) {
        this['quota:max_pps'] = quotaMaxPps;
        return this;
    };
    Object.defineProperty(FlavorExtraSpec.prototype, "quotaMaxPps", {
        get: function () {
            return this['quota:max_pps'];
        },
        set: function (quotaMaxPps) {
            this['quota:max_pps'] = quotaMaxPps;
        },
        enumerable: false,
        configurable: true
    });
    FlavorExtraSpec.prototype.withCondOperationCharge = function (condOperationCharge) {
        this['cond:operation:charge'] = condOperationCharge;
        return this;
    };
    Object.defineProperty(FlavorExtraSpec.prototype, "condOperationCharge", {
        get: function () {
            return this['cond:operation:charge'];
        },
        set: function (condOperationCharge) {
            this['cond:operation:charge'] = condOperationCharge;
        },
        enumerable: false,
        configurable: true
    });
    FlavorExtraSpec.prototype.withCondOperationChargeStop = function (condOperationChargeStop) {
        this['cond:operation:charge:stop'] = condOperationChargeStop;
        return this;
    };
    Object.defineProperty(FlavorExtraSpec.prototype, "condOperationChargeStop", {
        get: function () {
            return this['cond:operation:charge:stop'];
        },
        set: function (condOperationChargeStop) {
            this['cond:operation:charge:stop'] = condOperationChargeStop;
        },
        enumerable: false,
        configurable: true
    });
    FlavorExtraSpec.prototype.withCondSpotOperationAz = function (condSpotOperationAz) {
        this['cond:spot:operation:az'] = condSpotOperationAz;
        return this;
    };
    Object.defineProperty(FlavorExtraSpec.prototype, "condSpotOperationAz", {
        get: function () {
            return this['cond:spot:operation:az'];
        },
        set: function (condSpotOperationAz) {
            this['cond:spot:operation:az'] = condSpotOperationAz;
        },
        enumerable: false,
        configurable: true
    });
    FlavorExtraSpec.prototype.withCondOperationRoles = function (condOperationRoles) {
        this['cond:operation:roles'] = condOperationRoles;
        return this;
    };
    Object.defineProperty(FlavorExtraSpec.prototype, "condOperationRoles", {
        get: function () {
            return this['cond:operation:roles'];
        },
        set: function (condOperationRoles) {
            this['cond:operation:roles'] = condOperationRoles;
        },
        enumerable: false,
        configurable: true
    });
    FlavorExtraSpec.prototype.withCondSpotOperationStatus = function (condSpotOperationStatus) {
        this['cond:spot:operation:status'] = condSpotOperationStatus;
        return this;
    };
    Object.defineProperty(FlavorExtraSpec.prototype, "condSpotOperationStatus", {
        get: function () {
            return this['cond:spot:operation:status'];
        },
        set: function (condSpotOperationStatus) {
            this['cond:spot:operation:status'] = condSpotOperationStatus;
        },
        enumerable: false,
        configurable: true
    });
    FlavorExtraSpec.prototype.withCondNetwork = function (condNetwork) {
        this['cond:network'] = condNetwork;
        return this;
    };
    Object.defineProperty(FlavorExtraSpec.prototype, "condNetwork", {
        get: function () {
            return this['cond:network'];
        },
        set: function (condNetwork) {
            this['cond:network'] = condNetwork;
        },
        enumerable: false,
        configurable: true
    });
    FlavorExtraSpec.prototype.withCondStorage = function (condStorage) {
        this['cond:storage'] = condStorage;
        return this;
    };
    Object.defineProperty(FlavorExtraSpec.prototype, "condStorage", {
        get: function () {
            return this['cond:storage'];
        },
        set: function (condStorage) {
            this['cond:storage'] = condStorage;
        },
        enumerable: false,
        configurable: true
    });
    FlavorExtraSpec.prototype.withCondComputeLiveResizable = function (condComputeLiveResizable) {
        this['cond:compute:live_resizable'] = condComputeLiveResizable;
        return this;
    };
    Object.defineProperty(FlavorExtraSpec.prototype, "condComputeLiveResizable", {
        get: function () {
            return this['cond:compute:live_resizable'];
        },
        set: function (condComputeLiveResizable) {
            this['cond:compute:live_resizable'] = condComputeLiveResizable;
        },
        enumerable: false,
        configurable: true
    });
    FlavorExtraSpec.prototype.withCondCompute = function (condCompute) {
        this['cond:compute'] = condCompute;
        return this;
    };
    Object.defineProperty(FlavorExtraSpec.prototype, "condCompute", {
        get: function () {
            return this['cond:compute'];
        },
        set: function (condCompute) {
            this['cond:compute'] = condCompute;
        },
        enumerable: false,
        configurable: true
    });
    FlavorExtraSpec.prototype.withInfoGpuName = function (infoGpuName) {
        this['info:gpu:name'] = infoGpuName;
        return this;
    };
    Object.defineProperty(FlavorExtraSpec.prototype, "infoGpuName", {
        get: function () {
            return this['info:gpu:name'];
        },
        set: function (infoGpuName) {
            this['info:gpu:name'] = infoGpuName;
        },
        enumerable: false,
        configurable: true
    });
    FlavorExtraSpec.prototype.withInfoCpuName = function (infoCpuName) {
        this['info:cpu:name'] = infoCpuName;
        return this;
    };
    Object.defineProperty(FlavorExtraSpec.prototype, "infoCpuName", {
        get: function () {
            return this['info:cpu:name'];
        },
        set: function (infoCpuName) {
            this['info:cpu:name'] = infoCpuName;
        },
        enumerable: false,
        configurable: true
    });
    FlavorExtraSpec.prototype.withQuotaGpu = function (quotaGpu) {
        this['quota:gpu'] = quotaGpu;
        return this;
    };
    Object.defineProperty(FlavorExtraSpec.prototype, "quotaGpu", {
        get: function () {
            return this['quota:gpu'];
        },
        set: function (quotaGpu) {
            this['quota:gpu'] = quotaGpu;
        },
        enumerable: false,
        configurable: true
    });
    FlavorExtraSpec.prototype.withEcsInstanceArchitecture = function (ecsInstanceArchitecture) {
        this['ecs:instance_architecture'] = ecsInstanceArchitecture;
        return this;
    };
    Object.defineProperty(FlavorExtraSpec.prototype, "ecsInstanceArchitecture", {
        get: function () {
            return this['ecs:instance_architecture'];
        },
        set: function (ecsInstanceArchitecture) {
            this['ecs:instance_architecture'] = ecsInstanceArchitecture;
        },
        enumerable: false,
        configurable: true
    });
    return FlavorExtraSpec;
}());
exports.FlavorExtraSpec = FlavorExtraSpec;
