// @huaweicloud/huaweicloud-sdk-ecs/v2/model/Flavor.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Flavor = void 0;
var Flavor = /** @class */ (function () {
    function Flavor(id, name, vcpus, ram, disk, swap, oSFLVEXTDATAEphemeral, oSFLVDISABLEDDisabled, rxtxFactor, rxtxQuota, rxtxCap, osFlavorAccessIsPublic, links, osExtraSpecs) {
        this['id'] = id;
        this['name'] = name;
        this['vcpus'] = vcpus;
        this['ram'] = ram;
        this['disk'] = disk;
        this['swap'] = swap;
        this['OS-FLV-EXT-DATA:ephemeral'] = oSFLVEXTDATAEphemeral;
        this['OS-FLV-DISABLED:disabled'] = oSFLVDISABLEDDisabled;
        this['rxtx_factor'] = rxtxFactor;
        this['rxtx_quota'] = rxtxQuota;
        this['rxtx_cap'] = rxtxCap;
        this['os-flavor-access:is_public'] = osFlavorAccessIsPublic;
        this['links'] = links;
        this['os_extra_specs'] = osExtraSpecs;
    }
    Flavor.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    Flavor.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    Flavor.prototype.withVcpus = function (vcpus) {
        this['vcpus'] = vcpus;
        return this;
    };
    Flavor.prototype.withRam = function (ram) {
        this['ram'] = ram;
        return this;
    };
    Flavor.prototype.withDisk = function (disk) {
        this['disk'] = disk;
        return this;
    };
    Flavor.prototype.withSwap = function (swap) {
        this['swap'] = swap;
        return this;
    };
    Flavor.prototype.withOSFLVEXTDATAEphemeral = function (oSFLVEXTDATAEphemeral) {
        this['OS-FLV-EXT-DATA:ephemeral'] = oSFLVEXTDATAEphemeral;
        return this;
    };
    Object.defineProperty(Flavor.prototype, "oSFLVEXTDATAEphemeral", {
        get: function () {
            return this['OS-FLV-EXT-DATA:ephemeral'];
        },
        set: function (oSFLVEXTDATAEphemeral) {
            this['OS-FLV-EXT-DATA:ephemeral'] = oSFLVEXTDATAEphemeral;
        },
        enumerable: false,
        configurable: true
    });
    Flavor.prototype.withOSFLVDISABLEDDisabled = function (oSFLVDISABLEDDisabled) {
        this['OS-FLV-DISABLED:disabled'] = oSFLVDISABLEDDisabled;
        return this;
    };
    Object.defineProperty(Flavor.prototype, "oSFLVDISABLEDDisabled", {
        get: function () {
            return this['OS-FLV-DISABLED:disabled'];
        },
        set: function (oSFLVDISABLEDDisabled) {
            this['OS-FLV-DISABLED:disabled'] = oSFLVDISABLEDDisabled;
        },
        enumerable: false,
        configurable: true
    });
    Flavor.prototype.withRxtxFactor = function (rxtxFactor) {
        this['rxtx_factor'] = rxtxFactor;
        return this;
    };
    Object.defineProperty(Flavor.prototype, "rxtxFactor", {
        get: function () {
            return this['rxtx_factor'];
        },
        set: function (rxtxFactor) {
            this['rxtx_factor'] = rxtxFactor;
        },
        enumerable: false,
        configurable: true
    });
    Flavor.prototype.withRxtxQuota = function (rxtxQuota) {
        this['rxtx_quota'] = rxtxQuota;
        return this;
    };
    Object.defineProperty(Flavor.prototype, "rxtxQuota", {
        get: function () {
            return this['rxtx_quota'];
        },
        set: function (rxtxQuota) {
            this['rxtx_quota'] = rxtxQuota;
        },
        enumerable: false,
        configurable: true
    });
    Flavor.prototype.withRxtxCap = function (rxtxCap) {
        this['rxtx_cap'] = rxtxCap;
        return this;
    };
    Object.defineProperty(Flavor.prototype, "rxtxCap", {
        get: function () {
            return this['rxtx_cap'];
        },
        set: function (rxtxCap) {
            this['rxtx_cap'] = rxtxCap;
        },
        enumerable: false,
        configurable: true
    });
    Flavor.prototype.withOsFlavorAccessIsPublic = function (osFlavorAccessIsPublic) {
        this['os-flavor-access:is_public'] = osFlavorAccessIsPublic;
        return this;
    };
    Object.defineProperty(Flavor.prototype, "osFlavorAccessIsPublic", {
        get: function () {
            return this['os-flavor-access:is_public'];
        },
        set: function (osFlavorAccessIsPublic) {
            this['os-flavor-access:is_public'] = osFlavorAccessIsPublic;
        },
        enumerable: false,
        configurable: true
    });
    Flavor.prototype.withLinks = function (links) {
        this['links'] = links;
        return this;
    };
    Flavor.prototype.withOsExtraSpecs = function (osExtraSpecs) {
        this['os_extra_specs'] = osExtraSpecs;
        return this;
    };
    Object.defineProperty(Flavor.prototype, "osExtraSpecs", {
        get: function () {
            return this['os_extra_specs'];
        },
        set: function (osExtraSpecs) {
            this['os_extra_specs'] = osExtraSpecs;
        },
        enumerable: false,
        configurable: true
    });
    Flavor.prototype.withAttachableQuantity = function (attachableQuantity) {
        this['attachableQuantity'] = attachableQuantity;
        return this;
    };
    return Flavor;
}());
exports.Flavor = Flavor;
