// @huaweicloud/huaweicloud-sdk-ecs/v2/model/GetServerRemoteConsoleOption.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetServerRemoteConsoleOptionTypeEnum = exports.GetServerRemoteConsoleOptionProtocolEnum = exports.GetServerRemoteConsoleOption = void 0;
var GetServerRemoteConsoleOption = /** @class */ (function () {
    function GetServerRemoteConsoleOption(protocol, type) {
        this['protocol'] = protocol;
        this['type'] = type;
    }
    GetServerRemoteConsoleOption.prototype.withProtocol = function (protocol) {
        this['protocol'] = protocol;
        return this;
    };
    GetServerRemoteConsoleOption.prototype.withType = function (type) {
        this['type'] = type;
        return this;
    };
    return GetServerRemoteConsoleOption;
}());
exports.GetServerRemoteConsoleOption = GetServerRemoteConsoleOption;
/**
    * @export
    * @enum {string}
    */
var GetServerRemoteConsoleOptionProtocolEnum;
(function (GetServerRemoteConsoleOptionProtocolEnum) {
    GetServerRemoteConsoleOptionProtocolEnum["VNC"] = "vnc";
})(GetServerRemoteConsoleOptionProtocolEnum = exports.GetServerRemoteConsoleOptionProtocolEnum || (exports.GetServerRemoteConsoleOptionProtocolEnum = {}));
/**
    * @export
    * @enum {string}
    */
var GetServerRemoteConsoleOptionTypeEnum;
(function (GetServerRemoteConsoleOptionTypeEnum) {
    GetServerRemoteConsoleOptionTypeEnum["NOVNC"] = "novnc";
})(GetServerRemoteConsoleOptionTypeEnum = exports.GetServerRemoteConsoleOptionTypeEnum || (exports.GetServerRemoteConsoleOptionTypeEnum = {}));
