// @huaweicloud/huaweicloud-sdk-ecs/v2/model/ListServerGroupsResult.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListServerGroupsResult = void 0;
var ListServerGroupsResult = /** @class */ (function () {
    function ListServerGroupsResult(id, members, metadata, name, policies) {
        this['id'] = id;
        this['members'] = members;
        this['metadata'] = metadata;
        this['name'] = name;
        this['policies'] = policies;
    }
    ListServerGroupsResult.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    ListServerGroupsResult.prototype.withMembers = function (members) {
        this['members'] = members;
        return this;
    };
    ListServerGroupsResult.prototype.withMetadata = function (metadata) {
        this['metadata'] = metadata;
        return this;
    };
    ListServerGroupsResult.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    ListServerGroupsResult.prototype.withPolicies = function (policies) {
        this['policies'] = policies;
        return this;
    };
    return ListServerGroupsResult;
}());
exports.ListServerGroupsResult = ListServerGroupsResult;
