// @huaweicloud/huaweicloud-sdk-ecs/v2/model/ListResizeFlavorsResult.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListResizeFlavorsResult = void 0;
var ListResizeFlavorsResult = /** @class */ (function () {
    function ListResizeFlavorsResult(id, name, vcpus, ram, disk, swap, oSFLVEXTDATAEphemeral, oSFLVDISABLEDDisabled, rxtxFactor, rxtxQuota, rxtxCap, osFlavorAccessIsPublic, links, extraSpecs, instanceQuota) {
        this['id'] = id;
        this['name'] = name;
        this['vcpus'] = vcpus;
        this['ram'] = ram;
        this['disk'] = disk;
        this['swap'] = swap;
        this['OS-FLV-EXT-DATA:ephemeral'] = oSFLVEXTDATAEphemeral;
        this['OS-FLV-DISABLED:disabled'] = oSFLVDISABLEDDisabled;
        this['rxtx_factor'] = rxtxFactor;
        this['rxtx_quota'] = rxtxQuota;
        this['rxtx_cap'] = rxtxCap;
        this['os-flavor-access:is_public'] = osFlavorAccessIsPublic;
        this['links'] = links;
        this['extra_specs'] = extraSpecs;
        this['instance_quota'] = instanceQuota;
    }
    ListResizeFlavorsResult.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    ListResizeFlavorsResult.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    ListResizeFlavorsResult.prototype.withVcpus = function (vcpus) {
        this['vcpus'] = vcpus;
        return this;
    };
    ListResizeFlavorsResult.prototype.withRam = function (ram) {
        this['ram'] = ram;
        return this;
    };
    ListResizeFlavorsResult.prototype.withDisk = function (disk) {
        this['disk'] = disk;
        return this;
    };
    ListResizeFlavorsResult.prototype.withSwap = function (swap) {
        this['swap'] = swap;
        return this;
    };
    ListResizeFlavorsResult.prototype.withOSFLVEXTDATAEphemeral = function (oSFLVEXTDATAEphemeral) {
        this['OS-FLV-EXT-DATA:ephemeral'] = oSFLVEXTDATAEphemeral;
        return this;
    };
    Object.defineProperty(ListResizeFlavorsResult.prototype, "oSFLVEXTDATAEphemeral", {
        get: function () {
            return this['OS-FLV-EXT-DATA:ephemeral'];
        },
        set: function (oSFLVEXTDATAEphemeral) {
            this['OS-FLV-EXT-DATA:ephemeral'] = oSFLVEXTDATAEphemeral;
        },
        enumerable: false,
        configurable: true
    });
    ListResizeFlavorsResult.prototype.withOSFLVDISABLEDDisabled = function (oSFLVDISABLEDDisabled) {
        this['OS-FLV-DISABLED:disabled'] = oSFLVDISABLEDDisabled;
        return this;
    };
    Object.defineProperty(ListResizeFlavorsResult.prototype, "oSFLVDISABLEDDisabled", {
        get: function () {
            return this['OS-FLV-DISABLED:disabled'];
        },
        set: function (oSFLVDISABLEDDisabled) {
            this['OS-FLV-DISABLED:disabled'] = oSFLVDISABLEDDisabled;
        },
        enumerable: false,
        configurable: true
    });
    ListResizeFlavorsResult.prototype.withRxtxFactor = function (rxtxFactor) {
        this['rxtx_factor'] = rxtxFactor;
        return this;
    };
    Object.defineProperty(ListResizeFlavorsResult.prototype, "rxtxFactor", {
        get: function () {
            return this['rxtx_factor'];
        },
        set: function (rxtxFactor) {
            this['rxtx_factor'] = rxtxFactor;
        },
        enumerable: false,
        configurable: true
    });
    ListResizeFlavorsResult.prototype.withRxtxQuota = function (rxtxQuota) {
        this['rxtx_quota'] = rxtxQuota;
        return this;
    };
    Object.defineProperty(ListResizeFlavorsResult.prototype, "rxtxQuota", {
        get: function () {
            return this['rxtx_quota'];
        },
        set: function (rxtxQuota) {
            this['rxtx_quota'] = rxtxQuota;
        },
        enumerable: false,
        configurable: true
    });
    ListResizeFlavorsResult.prototype.withRxtxCap = function (rxtxCap) {
        this['rxtx_cap'] = rxtxCap;
        return this;
    };
    Object.defineProperty(ListResizeFlavorsResult.prototype, "rxtxCap", {
        get: function () {
            return this['rxtx_cap'];
        },
        set: function (rxtxCap) {
            this['rxtx_cap'] = rxtxCap;
        },
        enumerable: false,
        configurable: true
    });
    ListResizeFlavorsResult.prototype.withOsFlavorAccessIsPublic = function (osFlavorAccessIsPublic) {
        this['os-flavor-access:is_public'] = osFlavorAccessIsPublic;
        return this;
    };
    Object.defineProperty(ListResizeFlavorsResult.prototype, "osFlavorAccessIsPublic", {
        get: function () {
            return this['os-flavor-access:is_public'];
        },
        set: function (osFlavorAccessIsPublic) {
            this['os-flavor-access:is_public'] = osFlavorAccessIsPublic;
        },
        enumerable: false,
        configurable: true
    });
    ListResizeFlavorsResult.prototype.withLinks = function (links) {
        this['links'] = links;
        return this;
    };
    ListResizeFlavorsResult.prototype.withExtraSpecs = function (extraSpecs) {
        this['extra_specs'] = extraSpecs;
        return this;
    };
    Object.defineProperty(ListResizeFlavorsResult.prototype, "extraSpecs", {
        get: function () {
            return this['extra_specs'];
        },
        set: function (extraSpecs) {
            this['extra_specs'] = extraSpecs;
        },
        enumerable: false,
        configurable: true
    });
    ListResizeFlavorsResult.prototype.withInstanceQuota = function (instanceQuota) {
        this['instance_quota'] = instanceQuota;
        return this;
    };
    Object.defineProperty(ListResizeFlavorsResult.prototype, "instanceQuota", {
        get: function () {
            return this['instance_quota'];
        },
        set: function (instanceQuota) {
            this['instance_quota'] = instanceQuota;
        },
        enumerable: false,
        configurable: true
    });
    return ListResizeFlavorsResult;
}());
exports.ListResizeFlavorsResult = ListResizeFlavorsResult;
