// @huaweicloud/huaweicloud-sdk-ecs/v2/model/ListResizeFlavorsRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListResizeFlavorsRequestSortKeyEnum = exports.ListResizeFlavorsRequestSortDirEnum = exports.ListResizeFlavorsRequest = void 0;
var ListResizeFlavorsRequest = /** @class */ (function () {
    function ListResizeFlavorsRequest() {
    }
    ListResizeFlavorsRequest.prototype.withInstanceUuid = function (instanceUuid) {
        this['instance_uuid'] = instanceUuid;
        return this;
    };
    Object.defineProperty(ListResizeFlavorsRequest.prototype, "instanceUuid", {
        get: function () {
            return this['instance_uuid'];
        },
        set: function (instanceUuid) {
            this['instance_uuid'] = instanceUuid;
        },
        enumerable: false,
        configurable: true
    });
    ListResizeFlavorsRequest.prototype.withLimit = function (limit) {
        this['limit'] = limit;
        return this;
    };
    ListResizeFlavorsRequest.prototype.withMarker = function (marker) {
        this['marker'] = marker;
        return this;
    };
    ListResizeFlavorsRequest.prototype.withSortDir = function (sortDir) {
        this['sort_dir'] = sortDir;
        return this;
    };
    Object.defineProperty(ListResizeFlavorsRequest.prototype, "sortDir", {
        get: function () {
            return this['sort_dir'];
        },
        set: function (sortDir) {
            this['sort_dir'] = sortDir;
        },
        enumerable: false,
        configurable: true
    });
    ListResizeFlavorsRequest.prototype.withSortKey = function (sortKey) {
        this['sort_key'] = sortKey;
        return this;
    };
    Object.defineProperty(ListResizeFlavorsRequest.prototype, "sortKey", {
        get: function () {
            return this['sort_key'];
        },
        set: function (sortKey) {
            this['sort_key'] = sortKey;
        },
        enumerable: false,
        configurable: true
    });
    ListResizeFlavorsRequest.prototype.withSourceFlavorId = function (sourceFlavorId) {
        this['source_flavor_id'] = sourceFlavorId;
        return this;
    };
    Object.defineProperty(ListResizeFlavorsRequest.prototype, "sourceFlavorId", {
        get: function () {
            return this['source_flavor_id'];
        },
        set: function (sourceFlavorId) {
            this['source_flavor_id'] = sourceFlavorId;
        },
        enumerable: false,
        configurable: true
    });
    ListResizeFlavorsRequest.prototype.withSourceFlavorName = function (sourceFlavorName) {
        this['source_flavor_name'] = sourceFlavorName;
        return this;
    };
    Object.defineProperty(ListResizeFlavorsRequest.prototype, "sourceFlavorName", {
        get: function () {
            return this['source_flavor_name'];
        },
        set: function (sourceFlavorName) {
            this['source_flavor_name'] = sourceFlavorName;
        },
        enumerable: false,
        configurable: true
    });
    return ListResizeFlavorsRequest;
}());
exports.ListResizeFlavorsRequest = ListResizeFlavorsRequest;
/**
    * @export
    * @enum {string}
    */
var ListResizeFlavorsRequestSortDirEnum;
(function (ListResizeFlavorsRequestSortDirEnum) {
    ListResizeFlavorsRequestSortDirEnum["ASC"] = "asc";
    ListResizeFlavorsRequestSortDirEnum["DESC"] = "desc";
})(ListResizeFlavorsRequestSortDirEnum = exports.ListResizeFlavorsRequestSortDirEnum || (exports.ListResizeFlavorsRequestSortDirEnum = {}));
/**
    * @export
    * @enum {string}
    */
var ListResizeFlavorsRequestSortKeyEnum;
(function (ListResizeFlavorsRequestSortKeyEnum) {
    ListResizeFlavorsRequestSortKeyEnum["FLAVORID"] = "flavorid";
    ListResizeFlavorsRequestSortKeyEnum["SORT_KEY"] = "sort_key";
    ListResizeFlavorsRequestSortKeyEnum["NAME"] = "name";
    ListResizeFlavorsRequestSortKeyEnum["MEMORY_MB"] = "memory_mb";
    ListResizeFlavorsRequestSortKeyEnum["VCPUS"] = "vcpus";
    ListResizeFlavorsRequestSortKeyEnum["ROOT_GB"] = "root_gb";
})(ListResizeFlavorsRequestSortKeyEnum = exports.ListResizeFlavorsRequestSortKeyEnum || (exports.ListResizeFlavorsRequestSortKeyEnum = {}));
