// @huaweicloud/huaweicloud-sdk-ecs/v2/model/ListServersDetailsRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListServersDetailsRequest = void 0;
var ListServersDetailsRequest = /** @class */ (function () {
    function ListServersDetailsRequest() {
    }
    ListServersDetailsRequest.prototype.withEnterpriseProjectId = function (enterpriseProjectId) {
        this['enterprise_project_id'] = enterpriseProjectId;
        return this;
    };
    Object.defineProperty(ListServersDetailsRequest.prototype, "enterpriseProjectId", {
        get: function () {
            return this['enterprise_project_id'];
        },
        set: function (enterpriseProjectId) {
            this['enterprise_project_id'] = enterpriseProjectId;
        },
        enumerable: false,
        configurable: true
    });
    ListServersDetailsRequest.prototype.withFlavor = function (flavor) {
        this['flavor'] = flavor;
        return this;
    };
    ListServersDetailsRequest.prototype.withIp = function (ip) {
        this['ip'] = ip;
        return this;
    };
    ListServersDetailsRequest.prototype.withLimit = function (limit) {
        this['limit'] = limit;
        return this;
    };
    ListServersDetailsRequest.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    ListServersDetailsRequest.prototype.withNotTags = function (notTags) {
        this['not-tags'] = notTags;
        return this;
    };
    Object.defineProperty(ListServersDetailsRequest.prototype, "notTags", {
        get: function () {
            return this['not-tags'];
        },
        set: function (notTags) {
            this['not-tags'] = notTags;
        },
        enumerable: false,
        configurable: true
    });
    ListServersDetailsRequest.prototype.withOffset = function (offset) {
        this['offset'] = offset;
        return this;
    };
    ListServersDetailsRequest.prototype.withReservationId = function (reservationId) {
        this['reservation_id'] = reservationId;
        return this;
    };
    Object.defineProperty(ListServersDetailsRequest.prototype, "reservationId", {
        get: function () {
            return this['reservation_id'];
        },
        set: function (reservationId) {
            this['reservation_id'] = reservationId;
        },
        enumerable: false,
        configurable: true
    });
    ListServersDetailsRequest.prototype.withStatus = function (status) {
        this['status'] = status;
        return this;
    };
    ListServersDetailsRequest.prototype.withTags = function (tags) {
        this['tags'] = tags;
        return this;
    };
    return ListServersDetailsRequest;
}());
exports.ListServersDetailsRequest = ListServersDetailsRequest;
