// @huaweicloud/huaweicloud-sdk-ecs/v2/model/NovaServerFlavor.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NovaServerFlavor = void 0;
var NovaServerFlavor = /** @class */ (function () {
    function NovaServerFlavor() {
    }
    NovaServerFlavor.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    NovaServerFlavor.prototype.withLinks = function (links) {
        this['links'] = links;
        return this;
    };
    NovaServerFlavor.prototype.withVcpus = function (vcpus) {
        this['vcpus'] = vcpus;
        return this;
    };
    NovaServerFlavor.prototype.withRam = function (ram) {
        this['ram'] = ram;
        return this;
    };
    NovaServerFlavor.prototype.withDisk = function (disk) {
        this['disk'] = disk;
        return this;
    };
    NovaServerFlavor.prototype.withEphemeral = function (ephemeral) {
        this['ephemeral'] = ephemeral;
        return this;
    };
    NovaServerFlavor.prototype.withSwap = function (swap) {
        this['swap'] = swap;
        return this;
    };
    NovaServerFlavor.prototype.withOriginalName = function (originalName) {
        this['original_name'] = originalName;
        return this;
    };
    Object.defineProperty(NovaServerFlavor.prototype, "originalName", {
        get: function () {
            return this['original_name'];
        },
        set: function (originalName) {
            this['original_name'] = originalName;
        },
        enumerable: false,
        configurable: true
    });
    NovaServerFlavor.prototype.withExtraSpecs = function (extraSpecs) {
        this['extra_specs'] = extraSpecs;
        return this;
    };
    Object.defineProperty(NovaServerFlavor.prototype, "extraSpecs", {
        get: function () {
            return this['extra_specs'];
        },
        set: function (extraSpecs) {
            this['extra_specs'] = extraSpecs;
        },
        enumerable: false,
        configurable: true
    });
    return NovaServerFlavor;
}());
exports.NovaServerFlavor = NovaServerFlavor;
