// @huaweicloud/huaweicloud-sdk-ecs/v2/model/NovaCreateKeypairOption.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NovaCreateKeypairOptionTypeEnum = exports.NovaCreateKeypairOption = void 0;
var NovaCreateKeypairOption = /** @class */ (function () {
    function NovaCreateKeypairOption(name) {
        this['name'] = name;
    }
    NovaCreateKeypairOption.prototype.withPublicKey = function (publicKey) {
        this['public_key'] = publicKey;
        return this;
    };
    Object.defineProperty(NovaCreateKeypairOption.prototype, "publicKey", {
        get: function () {
            return this['public_key'];
        },
        set: function (publicKey) {
            this['public_key'] = publicKey;
        },
        enumerable: false,
        configurable: true
    });
    NovaCreateKeypairOption.prototype.withType = function (type) {
        this['type'] = type;
        return this;
    };
    NovaCreateKeypairOption.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    NovaCreateKeypairOption.prototype.withUserId = function (userId) {
        this['user_id'] = userId;
        return this;
    };
    Object.defineProperty(NovaCreateKeypairOption.prototype, "userId", {
        get: function () {
            return this['user_id'];
        },
        set: function (userId) {
            this['user_id'] = userId;
        },
        enumerable: false,
        configurable: true
    });
    return NovaCreateKeypairOption;
}());
exports.NovaCreateKeypairOption = NovaCreateKeypairOption;
/**
    * @export
    * @enum {string}
    */
var NovaCreateKeypairOptionTypeEnum;
(function (NovaCreateKeypairOptionTypeEnum) {
    NovaCreateKeypairOptionTypeEnum["SSH"] = "ssh";
    NovaCreateKeypairOptionTypeEnum["X509"] = "x509";
})(NovaCreateKeypairOptionTypeEnum = exports.NovaCreateKeypairOptionTypeEnum || (exports.NovaCreateKeypairOptionTypeEnum = {}));
