// @huaweicloud/huaweicloud-sdk-ecs/v2/model/NovaServerBlockDeviceMapping.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NovaServerBlockDeviceMappingDestinationTypeEnum = exports.NovaServerBlockDeviceMappingSourceTypeEnum = exports.NovaServerBlockDeviceMapping = void 0;
var NovaServerBlockDeviceMapping = /** @class */ (function () {
    function NovaServerBlockDeviceMapping(sourceType) {
        this['source_type'] = sourceType;
    }
    NovaServerBlockDeviceMapping.prototype.withSourceType = function (sourceType) {
        this['source_type'] = sourceType;
        return this;
    };
    Object.defineProperty(NovaServerBlockDeviceMapping.prototype, "sourceType", {
        get: function () {
            return this['source_type'];
        },
        set: function (sourceType) {
            this['source_type'] = sourceType;
        },
        enumerable: false,
        configurable: true
    });
    NovaServerBlockDeviceMapping.prototype.withDestinationType = function (destinationType) {
        this['destination_type'] = destinationType;
        return this;
    };
    Object.defineProperty(NovaServerBlockDeviceMapping.prototype, "destinationType", {
        get: function () {
            return this['destination_type'];
        },
        set: function (destinationType) {
            this['destination_type'] = destinationType;
        },
        enumerable: false,
        configurable: true
    });
    NovaServerBlockDeviceMapping.prototype.withGuestFormat = function (guestFormat) {
        this['guest_format'] = guestFormat;
        return this;
    };
    Object.defineProperty(NovaServerBlockDeviceMapping.prototype, "guestFormat", {
        get: function () {
            return this['guest_format'];
        },
        set: function (guestFormat) {
            this['guest_format'] = guestFormat;
        },
        enumerable: false,
        configurable: true
    });
    NovaServerBlockDeviceMapping.prototype.withDeviceName = function (deviceName) {
        this['device_name'] = deviceName;
        return this;
    };
    Object.defineProperty(NovaServerBlockDeviceMapping.prototype, "deviceName", {
        get: function () {
            return this['device_name'];
        },
        set: function (deviceName) {
            this['device_name'] = deviceName;
        },
        enumerable: false,
        configurable: true
    });
    NovaServerBlockDeviceMapping.prototype.withDeleteOnTermination = function (deleteOnTermination) {
        this['delete_on_termination'] = deleteOnTermination;
        return this;
    };
    Object.defineProperty(NovaServerBlockDeviceMapping.prototype, "deleteOnTermination", {
        get: function () {
            return this['delete_on_termination'];
        },
        set: function (deleteOnTermination) {
            this['delete_on_termination'] = deleteOnTermination;
        },
        enumerable: false,
        configurable: true
    });
    NovaServerBlockDeviceMapping.prototype.withBootIndex = function (bootIndex) {
        this['boot_index'] = bootIndex;
        return this;
    };
    Object.defineProperty(NovaServerBlockDeviceMapping.prototype, "bootIndex", {
        get: function () {
            return this['boot_index'];
        },
        set: function (bootIndex) {
            this['boot_index'] = bootIndex;
        },
        enumerable: false,
        configurable: true
    });
    NovaServerBlockDeviceMapping.prototype.withUuid = function (uuid) {
        this['uuid'] = uuid;
        return this;
    };
    NovaServerBlockDeviceMapping.prototype.withVolumeSize = function (volumeSize) {
        this['volume_size'] = volumeSize;
        return this;
    };
    Object.defineProperty(NovaServerBlockDeviceMapping.prototype, "volumeSize", {
        get: function () {
            return this['volume_size'];
        },
        set: function (volumeSize) {
            this['volume_size'] = volumeSize;
        },
        enumerable: false,
        configurable: true
    });
    NovaServerBlockDeviceMapping.prototype.withVolumeType = function (volumeType) {
        this['volume_type'] = volumeType;
        return this;
    };
    Object.defineProperty(NovaServerBlockDeviceMapping.prototype, "volumeType", {
        get: function () {
            return this['volume_type'];
        },
        set: function (volumeType) {
            this['volume_type'] = volumeType;
        },
        enumerable: false,
        configurable: true
    });
    return NovaServerBlockDeviceMapping;
}());
exports.NovaServerBlockDeviceMapping = NovaServerBlockDeviceMapping;
/**
    * @export
    * @enum {string}
    */
var NovaServerBlockDeviceMappingSourceTypeEnum;
(function (NovaServerBlockDeviceMappingSourceTypeEnum) {
    NovaServerBlockDeviceMappingSourceTypeEnum["BLANK"] = "blank";
    NovaServerBlockDeviceMappingSourceTypeEnum["SNAPSHOT"] = "snapshot";
    NovaServerBlockDeviceMappingSourceTypeEnum["VOLUME"] = "volume";
    NovaServerBlockDeviceMappingSourceTypeEnum["IMAGE"] = "image";
})(NovaServerBlockDeviceMappingSourceTypeEnum = exports.NovaServerBlockDeviceMappingSourceTypeEnum || (exports.NovaServerBlockDeviceMappingSourceTypeEnum = {}));
/**
    * @export
    * @enum {string}
    */
var NovaServerBlockDeviceMappingDestinationTypeEnum;
(function (NovaServerBlockDeviceMappingDestinationTypeEnum) {
    NovaServerBlockDeviceMappingDestinationTypeEnum["VOLUME"] = "volume";
})(NovaServerBlockDeviceMappingDestinationTypeEnum = exports.NovaServerBlockDeviceMappingDestinationTypeEnum || (exports.NovaServerBlockDeviceMappingDestinationTypeEnum = {}));
