// @huaweicloud/huaweicloud-sdk-ecs/v2/model/NovaNetwork.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NovaNetwork = void 0;
var NovaNetwork = /** @class */ (function () {
    function NovaNetwork(addr, version, oSEXTIPSMACMacAddr, oSEXTIPSType) {
        this['addr'] = addr;
        this['version'] = version;
        this['OS-EXT-IPS-MAC:mac_addr'] = oSEXTIPSMACMacAddr;
        this['OS-EXT-IPS:type'] = oSEXTIPSType;
    }
    NovaNetwork.prototype.withAddr = function (addr) {
        this['addr'] = addr;
        return this;
    };
    NovaNetwork.prototype.withVersion = function (version) {
        this['version'] = version;
        return this;
    };
    NovaNetwork.prototype.withOSEXTIPSMACMacAddr = function (oSEXTIPSMACMacAddr) {
        this['OS-EXT-IPS-MAC:mac_addr'] = oSEXTIPSMACMacAddr;
        return this;
    };
    Object.defineProperty(NovaNetwork.prototype, "oSEXTIPSMACMacAddr", {
        get: function () {
            return this['OS-EXT-IPS-MAC:mac_addr'];
        },
        set: function (oSEXTIPSMACMacAddr) {
            this['OS-EXT-IPS-MAC:mac_addr'] = oSEXTIPSMACMacAddr;
        },
        enumerable: false,
        configurable: true
    });
    NovaNetwork.prototype.withOSEXTIPSType = function (oSEXTIPSType) {
        this['OS-EXT-IPS:type'] = oSEXTIPSType;
        return this;
    };
    Object.defineProperty(NovaNetwork.prototype, "oSEXTIPSType", {
        get: function () {
            return this['OS-EXT-IPS:type'];
        },
        set: function (oSEXTIPSType) {
            this['OS-EXT-IPS:type'] = oSEXTIPSType;
        },
        enumerable: false,
        configurable: true
    });
    return NovaNetwork;
}());
exports.NovaNetwork = NovaNetwork;
