// @huaweicloud/huaweicloud-sdk-ecs/v2/model/NovaCreateServersResult.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NovaCreateServersResultOSDCFDiskConfigEnum = exports.NovaCreateServersResult = void 0;
var NovaCreateServersResult = /** @class */ (function () {
    function NovaCreateServersResult(id, links, securityGroups, oSDCFDiskConfig, adminPass) {
        this['id'] = id;
        this['links'] = links;
        this['security_groups'] = securityGroups;
        this['OS-DCF:diskConfig'] = oSDCFDiskConfig;
        this['adminPass'] = adminPass;
    }
    NovaCreateServersResult.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    NovaCreateServersResult.prototype.withLinks = function (links) {
        this['links'] = links;
        return this;
    };
    NovaCreateServersResult.prototype.withSecurityGroups = function (securityGroups) {
        this['security_groups'] = securityGroups;
        return this;
    };
    Object.defineProperty(NovaCreateServersResult.prototype, "securityGroups", {
        get: function () {
            return this['security_groups'];
        },
        set: function (securityGroups) {
            this['security_groups'] = securityGroups;
        },
        enumerable: false,
        configurable: true
    });
    NovaCreateServersResult.prototype.withOSDCFDiskConfig = function (oSDCFDiskConfig) {
        this['OS-DCF:diskConfig'] = oSDCFDiskConfig;
        return this;
    };
    Object.defineProperty(NovaCreateServersResult.prototype, "oSDCFDiskConfig", {
        get: function () {
            return this['OS-DCF:diskConfig'];
        },
        set: function (oSDCFDiskConfig) {
            this['OS-DCF:diskConfig'] = oSDCFDiskConfig;
        },
        enumerable: false,
        configurable: true
    });
    NovaCreateServersResult.prototype.withReservationId = function (reservationId) {
        this['reservation_id'] = reservationId;
        return this;
    };
    Object.defineProperty(NovaCreateServersResult.prototype, "reservationId", {
        get: function () {
            return this['reservation_id'];
        },
        set: function (reservationId) {
            this['reservation_id'] = reservationId;
        },
        enumerable: false,
        configurable: true
    });
    NovaCreateServersResult.prototype.withAdminPass = function (adminPass) {
        this['adminPass'] = adminPass;
        return this;
    };
    return NovaCreateServersResult;
}());
exports.NovaCreateServersResult = NovaCreateServersResult;
/**
    * @export
    * @enum {string}
    */
var NovaCreateServersResultOSDCFDiskConfigEnum;
(function (NovaCreateServersResultOSDCFDiskConfigEnum) {
    NovaCreateServersResultOSDCFDiskConfigEnum["MANUAL"] = "MANUAL";
    NovaCreateServersResultOSDCFDiskConfigEnum["AUTO"] = "AUTO";
})(NovaCreateServersResultOSDCFDiskConfigEnum = exports.NovaCreateServersResultOSDCFDiskConfigEnum || (exports.NovaCreateServersResultOSDCFDiskConfigEnum = {}));
