// @huaweicloud/huaweicloud-sdk-ecs/v2/model/NovaListServersDetailsRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NovaListServersDetailsRequestStatusEnum = exports.NovaListServersDetailsRequestSortKeyEnum = exports.NovaListServersDetailsRequest = void 0;
var NovaListServersDetailsRequest = /** @class */ (function () {
    function NovaListServersDetailsRequest() {
    }
    NovaListServersDetailsRequest.prototype.withChangesSince = function (changesSince) {
        this['changes-since'] = changesSince;
        return this;
    };
    Object.defineProperty(NovaListServersDetailsRequest.prototype, "changesSince", {
        get: function () {
            return this['changes-since'];
        },
        set: function (changesSince) {
            this['changes-since'] = changesSince;
        },
        enumerable: false,
        configurable: true
    });
    NovaListServersDetailsRequest.prototype.withFlavor = function (flavor) {
        this['flavor'] = flavor;
        return this;
    };
    NovaListServersDetailsRequest.prototype.withImage = function (image) {
        this['image'] = image;
        return this;
    };
    NovaListServersDetailsRequest.prototype.withIp = function (ip) {
        this['ip'] = ip;
        return this;
    };
    NovaListServersDetailsRequest.prototype.withLimit = function (limit) {
        this['limit'] = limit;
        return this;
    };
    NovaListServersDetailsRequest.prototype.withMarker = function (marker) {
        this['marker'] = marker;
        return this;
    };
    NovaListServersDetailsRequest.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    NovaListServersDetailsRequest.prototype.withNotTags = function (notTags) {
        this['not-tags'] = notTags;
        return this;
    };
    Object.defineProperty(NovaListServersDetailsRequest.prototype, "notTags", {
        get: function () {
            return this['not-tags'];
        },
        set: function (notTags) {
            this['not-tags'] = notTags;
        },
        enumerable: false,
        configurable: true
    });
    NovaListServersDetailsRequest.prototype.withReservationId = function (reservationId) {
        this['reservation_id'] = reservationId;
        return this;
    };
    Object.defineProperty(NovaListServersDetailsRequest.prototype, "reservationId", {
        get: function () {
            return this['reservation_id'];
        },
        set: function (reservationId) {
            this['reservation_id'] = reservationId;
        },
        enumerable: false,
        configurable: true
    });
    NovaListServersDetailsRequest.prototype.withSortKey = function (sortKey) {
        this['sort_key'] = sortKey;
        return this;
    };
    Object.defineProperty(NovaListServersDetailsRequest.prototype, "sortKey", {
        get: function () {
            return this['sort_key'];
        },
        set: function (sortKey) {
            this['sort_key'] = sortKey;
        },
        enumerable: false,
        configurable: true
    });
    NovaListServersDetailsRequest.prototype.withStatus = function (status) {
        this['status'] = status;
        return this;
    };
    NovaListServersDetailsRequest.prototype.withTags = function (tags) {
        this['tags'] = tags;
        return this;
    };
    NovaListServersDetailsRequest.prototype.withOpenStackAPIVersion = function (openStackAPIVersion) {
        this['OpenStack-API-Version'] = openStackAPIVersion;
        return this;
    };
    Object.defineProperty(NovaListServersDetailsRequest.prototype, "openStackAPIVersion", {
        get: function () {
            return this['OpenStack-API-Version'];
        },
        set: function (openStackAPIVersion) {
            this['OpenStack-API-Version'] = openStackAPIVersion;
        },
        enumerable: false,
        configurable: true
    });
    return NovaListServersDetailsRequest;
}());
exports.NovaListServersDetailsRequest = NovaListServersDetailsRequest;
/**
    * @export
    * @enum {string}
    */
var NovaListServersDetailsRequestSortKeyEnum;
(function (NovaListServersDetailsRequestSortKeyEnum) {
    NovaListServersDetailsRequestSortKeyEnum["CREATED_AT"] = "created_at";
    NovaListServersDetailsRequestSortKeyEnum["AVAILABILITY_ZONE"] = "availability_zone";
    NovaListServersDetailsRequestSortKeyEnum["DISPLAY_NAME"] = "display_name";
    NovaListServersDetailsRequestSortKeyEnum["HOST"] = "host";
    NovaListServersDetailsRequestSortKeyEnum["INSTANCE_TYPE_ID"] = "instance_type_id";
    NovaListServersDetailsRequestSortKeyEnum["KEY_NAME"] = "key_name";
    NovaListServersDetailsRequestSortKeyEnum["PROJECT_ID"] = "project_id";
    NovaListServersDetailsRequestSortKeyEnum["USER_ID"] = "user_id";
    NovaListServersDetailsRequestSortKeyEnum["UPDATED_AT"] = "updated_at";
    NovaListServersDetailsRequestSortKeyEnum["UUID"] = "uuid";
    NovaListServersDetailsRequestSortKeyEnum["VM_STATE"] = "vm_state";
})(NovaListServersDetailsRequestSortKeyEnum = exports.NovaListServersDetailsRequestSortKeyEnum || (exports.NovaListServersDetailsRequestSortKeyEnum = {}));
/**
    * @export
    * @enum {string}
    */
var NovaListServersDetailsRequestStatusEnum;
(function (NovaListServersDetailsRequestStatusEnum) {
    NovaListServersDetailsRequestStatusEnum["ACTIVE"] = "ACTIVE";
    NovaListServersDetailsRequestStatusEnum["BUILD"] = "BUILD";
    NovaListServersDetailsRequestStatusEnum["DELETED"] = "DELETED";
    NovaListServersDetailsRequestStatusEnum["ERROR"] = "ERROR";
    NovaListServersDetailsRequestStatusEnum["HARD_REBOOT"] = "HARD_REBOOT";
    NovaListServersDetailsRequestStatusEnum["MIGRATING"] = "MIGRATING";
    NovaListServersDetailsRequestStatusEnum["REBOOT"] = "REBOOT";
    NovaListServersDetailsRequestStatusEnum["RESIZE"] = "RESIZE";
    NovaListServersDetailsRequestStatusEnum["REVERT_RESIZE"] = "REVERT_RESIZE";
    NovaListServersDetailsRequestStatusEnum["SHELVED"] = "SHELVED";
    NovaListServersDetailsRequestStatusEnum["SHELVED_OFFLOADED"] = "SHELVED_OFFLOADED";
    NovaListServersDetailsRequestStatusEnum["SHUTOFF"] = "SHUTOFF";
    NovaListServersDetailsRequestStatusEnum["UNKNOWN"] = "UNKNOWN";
    NovaListServersDetailsRequestStatusEnum["VERIFY_RESIZE"] = "VERIFY_RESIZE";
})(NovaListServersDetailsRequestStatusEnum = exports.NovaListServersDetailsRequestStatusEnum || (exports.NovaListServersDetailsRequestStatusEnum = {}));
