// @huaweicloud/huaweicloud-sdk-ecs/v2/model/NovaServerSchedulerHints.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NovaServerSchedulerHintsTenancyEnum = exports.NovaServerSchedulerHints = void 0;
var NovaServerSchedulerHints = /** @class */ (function () {
    function NovaServerSchedulerHints() {
    }
    NovaServerSchedulerHints.prototype.withTenancy = function (tenancy) {
        this['tenancy'] = tenancy;
        return this;
    };
    NovaServerSchedulerHints.prototype.withDedicatedHostId = function (dedicatedHostId) {
        this['dedicated_host_id'] = dedicatedHostId;
        return this;
    };
    Object.defineProperty(NovaServerSchedulerHints.prototype, "dedicatedHostId", {
        get: function () {
            return this['dedicated_host_id'];
        },
        set: function (dedicatedHostId) {
            this['dedicated_host_id'] = dedicatedHostId;
        },
        enumerable: false,
        configurable: true
    });
    return NovaServerSchedulerHints;
}());
exports.NovaServerSchedulerHints = NovaServerSchedulerHints;
/**
    * @export
    * @enum {string}
    */
var NovaServerSchedulerHintsTenancyEnum;
(function (NovaServerSchedulerHintsTenancyEnum) {
    NovaServerSchedulerHintsTenancyEnum["SHARED"] = "shared";
    NovaServerSchedulerHintsTenancyEnum["DEDICATED"] = "dedicated";
})(NovaServerSchedulerHintsTenancyEnum = exports.NovaServerSchedulerHintsTenancyEnum || (exports.NovaServerSchedulerHintsTenancyEnum = {}));
