// @huaweicloud/huaweicloud-sdk-ecs/v2/model/NovaListKeypairsRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NovaListKeypairsRequest = void 0;
var NovaListKeypairsRequest = /** @class */ (function () {
    function NovaListKeypairsRequest() {
    }
    NovaListKeypairsRequest.prototype.withLimit = function (limit) {
        this['limit'] = limit;
        return this;
    };
    NovaListKeypairsRequest.prototype.withMarker = function (marker) {
        this['marker'] = marker;
        return this;
    };
    NovaListKeypairsRequest.prototype.withOpenStackAPIVersion = function (openStackAPIVersion) {
        this['OpenStack-API-Version'] = openStackAPIVersion;
        return this;
    };
    Object.defineProperty(NovaListKeypairsRequest.prototype, "openStackAPIVersion", {
        get: function () {
            return this['OpenStack-API-Version'];
        },
        set: function (openStackAPIVersion) {
            this['OpenStack-API-Version'] = openStackAPIVersion;
        },
        enumerable: false,
        configurable: true
    });
    return NovaListKeypairsRequest;
}());
exports.NovaListKeypairsRequest = NovaListKeypairsRequest;
