// @huaweicloud/huaweicloud-sdk-ecs/v2/model/NovaSecurityGroup.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NovaSecurityGroup = void 0;
var NovaSecurityGroup = /** @class */ (function () {
    function NovaSecurityGroup(description, id, name, tenantId, rules) {
        this['description'] = description;
        this['id'] = id;
        this['name'] = name;
        this['tenant_id'] = tenantId;
        this['rules'] = rules;
    }
    NovaSecurityGroup.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    NovaSecurityGroup.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    NovaSecurityGroup.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    NovaSecurityGroup.prototype.withTenantId = function (tenantId) {
        this['tenant_id'] = tenantId;
        return this;
    };
    Object.defineProperty(NovaSecurityGroup.prototype, "tenantId", {
        get: function () {
            return this['tenant_id'];
        },
        set: function (tenantId) {
            this['tenant_id'] = tenantId;
        },
        enumerable: false,
        configurable: true
    });
    NovaSecurityGroup.prototype.withRules = function (rules) {
        this['rules'] = rules;
        return this;
    };
    return NovaSecurityGroup;
}());
exports.NovaSecurityGroup = NovaSecurityGroup;
