// @huaweicloud/huaweicloud-sdk-ecs/v2/model/NovaKeypairDetail.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NovaKeypairDetail = void 0;
var NovaKeypairDetail = /** @class */ (function () {
    function NovaKeypairDetail(publicKey, name, fingerprint, createdAt, deleted, deletedAt, id, updatedAt, userId) {
        this['public_key'] = publicKey;
        this['name'] = name;
        this['fingerprint'] = fingerprint;
        this['created_at'] = createdAt;
        this['deleted'] = deleted;
        this['deleted_at'] = deletedAt;
        this['id'] = id;
        this['updated_at'] = updatedAt;
        this['user_id'] = userId;
    }
    NovaKeypairDetail.prototype.withPublicKey = function (publicKey) {
        this['public_key'] = publicKey;
        return this;
    };
    Object.defineProperty(NovaKeypairDetail.prototype, "publicKey", {
        get: function () {
            return this['public_key'];
        },
        set: function (publicKey) {
            this['public_key'] = publicKey;
        },
        enumerable: false,
        configurable: true
    });
    NovaKeypairDetail.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    NovaKeypairDetail.prototype.withFingerprint = function (fingerprint) {
        this['fingerprint'] = fingerprint;
        return this;
    };
    NovaKeypairDetail.prototype.withCreatedAt = function (createdAt) {
        this['created_at'] = createdAt;
        return this;
    };
    Object.defineProperty(NovaKeypairDetail.prototype, "createdAt", {
        get: function () {
            return this['created_at'];
        },
        set: function (createdAt) {
            this['created_at'] = createdAt;
        },
        enumerable: false,
        configurable: true
    });
    NovaKeypairDetail.prototype.withDeleted = function (deleted) {
        this['deleted'] = deleted;
        return this;
    };
    NovaKeypairDetail.prototype.withDeletedAt = function (deletedAt) {
        this['deleted_at'] = deletedAt;
        return this;
    };
    Object.defineProperty(NovaKeypairDetail.prototype, "deletedAt", {
        get: function () {
            return this['deleted_at'];
        },
        set: function (deletedAt) {
            this['deleted_at'] = deletedAt;
        },
        enumerable: false,
        configurable: true
    });
    NovaKeypairDetail.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    NovaKeypairDetail.prototype.withUpdatedAt = function (updatedAt) {
        this['updated_at'] = updatedAt;
        return this;
    };
    Object.defineProperty(NovaKeypairDetail.prototype, "updatedAt", {
        get: function () {
            return this['updated_at'];
        },
        set: function (updatedAt) {
            this['updated_at'] = updatedAt;
        },
        enumerable: false,
        configurable: true
    });
    NovaKeypairDetail.prototype.withUserId = function (userId) {
        this['user_id'] = userId;
        return this;
    };
    Object.defineProperty(NovaKeypairDetail.prototype, "userId", {
        get: function () {
            return this['user_id'];
        },
        set: function (userId) {
            this['user_id'] = userId;
        },
        enumerable: false,
        configurable: true
    });
    NovaKeypairDetail.prototype.withType = function (type) {
        this['type'] = type;
        return this;
    };
    return NovaKeypairDetail;
}());
exports.NovaKeypairDetail = NovaKeypairDetail;
