// @huaweicloud/huaweicloud-sdk-ecs/v2/model/NovaCreateServersOption.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NovaCreateServersOptionOSDCFDiskConfigEnum = exports.NovaCreateServersOption = void 0;
var NovaCreateServersOption = /** @class */ (function () {
    function NovaCreateServersOption(flavorRef, name, networks) {
        this['flavorRef'] = flavorRef;
        this['name'] = name;
        this['networks'] = networks;
    }
    NovaCreateServersOption.prototype.withAutoTerminateTime = function (autoTerminateTime) {
        this['auto_terminate_time'] = autoTerminateTime;
        return this;
    };
    Object.defineProperty(NovaCreateServersOption.prototype, "autoTerminateTime", {
        get: function () {
            return this['auto_terminate_time'];
        },
        set: function (autoTerminateTime) {
            this['auto_terminate_time'] = autoTerminateTime;
        },
        enumerable: false,
        configurable: true
    });
    NovaCreateServersOption.prototype.withImageRef = function (imageRef) {
        this['imageRef'] = imageRef;
        return this;
    };
    NovaCreateServersOption.prototype.withFlavorRef = function (flavorRef) {
        this['flavorRef'] = flavorRef;
        return this;
    };
    NovaCreateServersOption.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    NovaCreateServersOption.prototype.withMetadata = function (metadata) {
        this['metadata'] = metadata;
        return this;
    };
    NovaCreateServersOption.prototype.withAdminPass = function (adminPass) {
        this['adminPass'] = adminPass;
        return this;
    };
    NovaCreateServersOption.prototype.withBlockDeviceMappingV2 = function (blockDeviceMappingV2) {
        this['block_device_mapping_v2'] = blockDeviceMappingV2;
        return this;
    };
    Object.defineProperty(NovaCreateServersOption.prototype, "blockDeviceMappingV2", {
        get: function () {
            return this['block_device_mapping_v2'];
        },
        set: function (blockDeviceMappingV2) {
            this['block_device_mapping_v2'] = blockDeviceMappingV2;
        },
        enumerable: false,
        configurable: true
    });
    NovaCreateServersOption.prototype.withConfigDrive = function (configDrive) {
        this['config_drive'] = configDrive;
        return this;
    };
    Object.defineProperty(NovaCreateServersOption.prototype, "configDrive", {
        get: function () {
            return this['config_drive'];
        },
        set: function (configDrive) {
            this['config_drive'] = configDrive;
        },
        enumerable: false,
        configurable: true
    });
    NovaCreateServersOption.prototype.withSecurityGroups = function (securityGroups) {
        this['security_groups'] = securityGroups;
        return this;
    };
    Object.defineProperty(NovaCreateServersOption.prototype, "securityGroups", {
        get: function () {
            return this['security_groups'];
        },
        set: function (securityGroups) {
            this['security_groups'] = securityGroups;
        },
        enumerable: false,
        configurable: true
    });
    NovaCreateServersOption.prototype.withNetworks = function (networks) {
        this['networks'] = networks;
        return this;
    };
    NovaCreateServersOption.prototype.withKeyName = function (keyName) {
        this['key_name'] = keyName;
        return this;
    };
    Object.defineProperty(NovaCreateServersOption.prototype, "keyName", {
        get: function () {
            return this['key_name'];
        },
        set: function (keyName) {
            this['key_name'] = keyName;
        },
        enumerable: false,
        configurable: true
    });
    NovaCreateServersOption.prototype.withUserData = function (userData) {
        this['user_data'] = userData;
        return this;
    };
    Object.defineProperty(NovaCreateServersOption.prototype, "userData", {
        get: function () {
            return this['user_data'];
        },
        set: function (userData) {
            this['user_data'] = userData;
        },
        enumerable: false,
        configurable: true
    });
    NovaCreateServersOption.prototype.withAvailabilityZone = function (availabilityZone) {
        this['availability_zone'] = availabilityZone;
        return this;
    };
    Object.defineProperty(NovaCreateServersOption.prototype, "availabilityZone", {
        get: function () {
            return this['availability_zone'];
        },
        set: function (availabilityZone) {
            this['availability_zone'] = availabilityZone;
        },
        enumerable: false,
        configurable: true
    });
    NovaCreateServersOption.prototype.withReturnReservationId = function (returnReservationId) {
        this['return_reservation_id'] = returnReservationId;
        return this;
    };
    Object.defineProperty(NovaCreateServersOption.prototype, "returnReservationId", {
        get: function () {
            return this['return_reservation_id'];
        },
        set: function (returnReservationId) {
            this['return_reservation_id'] = returnReservationId;
        },
        enumerable: false,
        configurable: true
    });
    NovaCreateServersOption.prototype.withMinCount = function (minCount) {
        this['min_count'] = minCount;
        return this;
    };
    Object.defineProperty(NovaCreateServersOption.prototype, "minCount", {
        get: function () {
            return this['min_count'];
        },
        set: function (minCount) {
            this['min_count'] = minCount;
        },
        enumerable: false,
        configurable: true
    });
    NovaCreateServersOption.prototype.withMaxCount = function (maxCount) {
        this['max_count'] = maxCount;
        return this;
    };
    Object.defineProperty(NovaCreateServersOption.prototype, "maxCount", {
        get: function () {
            return this['max_count'];
        },
        set: function (maxCount) {
            this['max_count'] = maxCount;
        },
        enumerable: false,
        configurable: true
    });
    NovaCreateServersOption.prototype.withOSDCFDiskConfig = function (oSDCFDiskConfig) {
        this['OS-DCF:diskConfig'] = oSDCFDiskConfig;
        return this;
    };
    Object.defineProperty(NovaCreateServersOption.prototype, "oSDCFDiskConfig", {
        get: function () {
            return this['OS-DCF:diskConfig'];
        },
        set: function (oSDCFDiskConfig) {
            this['OS-DCF:diskConfig'] = oSDCFDiskConfig;
        },
        enumerable: false,
        configurable: true
    });
    NovaCreateServersOption.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    return NovaCreateServersOption;
}());
exports.NovaCreateServersOption = NovaCreateServersOption;
/**
    * @export
    * @enum {string}
    */
var NovaCreateServersOptionOSDCFDiskConfigEnum;
(function (NovaCreateServersOptionOSDCFDiskConfigEnum) {
    NovaCreateServersOptionOSDCFDiskConfigEnum["AUTO"] = "AUTO";
    NovaCreateServersOptionOSDCFDiskConfigEnum["MANUAL"] = "MANUAL";
})(NovaCreateServersOptionOSDCFDiskConfigEnum = exports.NovaCreateServersOptionOSDCFDiskConfigEnum || (exports.NovaCreateServersOptionOSDCFDiskConfigEnum = {}));
