// @huaweicloud/huaweicloud-sdk-ecs/v2/model/NovaSecurityGroupCommonRule.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NovaSecurityGroupCommonRule = void 0;
var NovaSecurityGroupCommonRule = /** @class */ (function () {
    function NovaSecurityGroupCommonRule(fromPort, group, id, ipProtocol, ipRange, parentGroupId, toPort) {
        this['from_port'] = fromPort;
        this['group'] = group;
        this['id'] = id;
        this['ip_protocol'] = ipProtocol;
        this['ip_range'] = ipRange;
        this['parent_group_id'] = parentGroupId;
        this['to_port'] = toPort;
    }
    NovaSecurityGroupCommonRule.prototype.withFromPort = function (fromPort) {
        this['from_port'] = fromPort;
        return this;
    };
    Object.defineProperty(NovaSecurityGroupCommonRule.prototype, "fromPort", {
        get: function () {
            return this['from_port'];
        },
        set: function (fromPort) {
            this['from_port'] = fromPort;
        },
        enumerable: false,
        configurable: true
    });
    NovaSecurityGroupCommonRule.prototype.withGroup = function (group) {
        this['group'] = group;
        return this;
    };
    NovaSecurityGroupCommonRule.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    NovaSecurityGroupCommonRule.prototype.withIpProtocol = function (ipProtocol) {
        this['ip_protocol'] = ipProtocol;
        return this;
    };
    Object.defineProperty(NovaSecurityGroupCommonRule.prototype, "ipProtocol", {
        get: function () {
            return this['ip_protocol'];
        },
        set: function (ipProtocol) {
            this['ip_protocol'] = ipProtocol;
        },
        enumerable: false,
        configurable: true
    });
    NovaSecurityGroupCommonRule.prototype.withIpRange = function (ipRange) {
        this['ip_range'] = ipRange;
        return this;
    };
    Object.defineProperty(NovaSecurityGroupCommonRule.prototype, "ipRange", {
        get: function () {
            return this['ip_range'];
        },
        set: function (ipRange) {
            this['ip_range'] = ipRange;
        },
        enumerable: false,
        configurable: true
    });
    NovaSecurityGroupCommonRule.prototype.withParentGroupId = function (parentGroupId) {
        this['parent_group_id'] = parentGroupId;
        return this;
    };
    Object.defineProperty(NovaSecurityGroupCommonRule.prototype, "parentGroupId", {
        get: function () {
            return this['parent_group_id'];
        },
        set: function (parentGroupId) {
            this['parent_group_id'] = parentGroupId;
        },
        enumerable: false,
        configurable: true
    });
    NovaSecurityGroupCommonRule.prototype.withToPort = function (toPort) {
        this['to_port'] = toPort;
        return this;
    };
    Object.defineProperty(NovaSecurityGroupCommonRule.prototype, "toPort", {
        get: function () {
            return this['to_port'];
        },
        set: function (toPort) {
            this['to_port'] = toPort;
        },
        enumerable: false,
        configurable: true
    });
    return NovaSecurityGroupCommonRule;
}());
exports.NovaSecurityGroupCommonRule = NovaSecurityGroupCommonRule;
