// @huaweicloud/huaweicloud-sdk-ecs/v2/model/NovaCreateServersSchedulerHint.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NovaCreateServersSchedulerHint = void 0;
var NovaCreateServersSchedulerHint = /** @class */ (function () {
    function NovaCreateServersSchedulerHint() {
    }
    NovaCreateServersSchedulerHint.prototype.withGroup = function (group) {
        this['group'] = group;
        return this;
    };
    NovaCreateServersSchedulerHint.prototype.withDifferentHost = function (differentHost) {
        this['different_host'] = differentHost;
        return this;
    };
    Object.defineProperty(NovaCreateServersSchedulerHint.prototype, "differentHost", {
        get: function () {
            return this['different_host'];
        },
        set: function (differentHost) {
            this['different_host'] = differentHost;
        },
        enumerable: false,
        configurable: true
    });
    NovaCreateServersSchedulerHint.prototype.withSameHost = function (sameHost) {
        this['same_host'] = sameHost;
        return this;
    };
    Object.defineProperty(NovaCreateServersSchedulerHint.prototype, "sameHost", {
        get: function () {
            return this['same_host'];
        },
        set: function (sameHost) {
            this['same_host'] = sameHost;
        },
        enumerable: false,
        configurable: true
    });
    NovaCreateServersSchedulerHint.prototype.withCidr = function (cidr) {
        this['cidr'] = cidr;
        return this;
    };
    NovaCreateServersSchedulerHint.prototype.withBuildNearHostIp = function (buildNearHostIp) {
        this['build_near_host_ip'] = buildNearHostIp;
        return this;
    };
    Object.defineProperty(NovaCreateServersSchedulerHint.prototype, "buildNearHostIp", {
        get: function () {
            return this['build_near_host_ip'];
        },
        set: function (buildNearHostIp) {
            this['build_near_host_ip'] = buildNearHostIp;
        },
        enumerable: false,
        configurable: true
    });
    NovaCreateServersSchedulerHint.prototype.withTenancy = function (tenancy) {
        this['tenancy'] = tenancy;
        return this;
    };
    NovaCreateServersSchedulerHint.prototype.withDedicatedHostId = function (dedicatedHostId) {
        this['dedicated_host_id'] = dedicatedHostId;
        return this;
    };
    Object.defineProperty(NovaCreateServersSchedulerHint.prototype, "dedicatedHostId", {
        get: function () {
            return this['dedicated_host_id'];
        },
        set: function (dedicatedHostId) {
            this['dedicated_host_id'] = dedicatedHostId;
        },
        enumerable: false,
        configurable: true
    });
    return NovaCreateServersSchedulerHint;
}());
exports.NovaCreateServersSchedulerHint = NovaCreateServersSchedulerHint;
