// @huaweicloud/huaweicloud-sdk-ecs/v2/model/NovaSimpleKeypair.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NovaSimpleKeypair = void 0;
var NovaSimpleKeypair = /** @class */ (function () {
    function NovaSimpleKeypair(fingerprint, name, publicKey) {
        this['fingerprint'] = fingerprint;
        this['name'] = name;
        this['public_key'] = publicKey;
    }
    NovaSimpleKeypair.prototype.withFingerprint = function (fingerprint) {
        this['fingerprint'] = fingerprint;
        return this;
    };
    NovaSimpleKeypair.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    NovaSimpleKeypair.prototype.withPublicKey = function (publicKey) {
        this['public_key'] = publicKey;
        return this;
    };
    Object.defineProperty(NovaSimpleKeypair.prototype, "publicKey", {
        get: function () {
            return this['public_key'];
        },
        set: function (publicKey) {
            this['public_key'] = publicKey;
        },
        enumerable: false,
        configurable: true
    });
    NovaSimpleKeypair.prototype.withType = function (type) {
        this['type'] = type;
        return this;
    };
    return NovaSimpleKeypair;
}());
exports.NovaSimpleKeypair = NovaSimpleKeypair;
