// @huaweicloud/huaweicloud-sdk-ecs/v2/model/PostPaidServerDataVolume.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PostPaidServerDataVolumeClusterTypeEnum = exports.PostPaidServerDataVolumeVolumetypeEnum = exports.PostPaidServerDataVolume = void 0;
var PostPaidServerDataVolume = /** @class */ (function () {
    function PostPaidServerDataVolume(volumetype, size) {
        this['volumetype'] = volumetype;
        this['size'] = size;
    }
    PostPaidServerDataVolume.prototype.withVolumetype = function (volumetype) {
        this['volumetype'] = volumetype;
        return this;
    };
    PostPaidServerDataVolume.prototype.withSize = function (size) {
        this['size'] = size;
        return this;
    };
    PostPaidServerDataVolume.prototype.withShareable = function (shareable) {
        this['shareable'] = shareable;
        return this;
    };
    PostPaidServerDataVolume.prototype.withMultiattach = function (multiattach) {
        this['multiattach'] = multiattach;
        return this;
    };
    PostPaidServerDataVolume.prototype.withHwPassthrough = function (hwPassthrough) {
        this['hw:passthrough'] = hwPassthrough;
        return this;
    };
    Object.defineProperty(PostPaidServerDataVolume.prototype, "hwPassthrough", {
        get: function () {
            return this['hw:passthrough'];
        },
        set: function (hwPassthrough) {
            this['hw:passthrough'] = hwPassthrough;
        },
        enumerable: false,
        configurable: true
    });
    PostPaidServerDataVolume.prototype.withExtendparam = function (extendparam) {
        this['extendparam'] = extendparam;
        return this;
    };
    PostPaidServerDataVolume.prototype.withClusterType = function (clusterType) {
        this['cluster_type'] = clusterType;
        return this;
    };
    Object.defineProperty(PostPaidServerDataVolume.prototype, "clusterType", {
        get: function () {
            return this['cluster_type'];
        },
        set: function (clusterType) {
            this['cluster_type'] = clusterType;
        },
        enumerable: false,
        configurable: true
    });
    PostPaidServerDataVolume.prototype.withClusterId = function (clusterId) {
        this['cluster_id'] = clusterId;
        return this;
    };
    Object.defineProperty(PostPaidServerDataVolume.prototype, "clusterId", {
        get: function () {
            return this['cluster_id'];
        },
        set: function (clusterId) {
            this['cluster_id'] = clusterId;
        },
        enumerable: false,
        configurable: true
    });
    PostPaidServerDataVolume.prototype.withMetadata = function (metadata) {
        this['metadata'] = metadata;
        return this;
    };
    PostPaidServerDataVolume.prototype.withDataImageId = function (dataImageId) {
        this['data_image_id'] = dataImageId;
        return this;
    };
    Object.defineProperty(PostPaidServerDataVolume.prototype, "dataImageId", {
        get: function () {
            return this['data_image_id'];
        },
        set: function (dataImageId) {
            this['data_image_id'] = dataImageId;
        },
        enumerable: false,
        configurable: true
    });
    return PostPaidServerDataVolume;
}());
exports.PostPaidServerDataVolume = PostPaidServerDataVolume;
/**
    * @export
    * @enum {string}
    */
var PostPaidServerDataVolumeVolumetypeEnum;
(function (PostPaidServerDataVolumeVolumetypeEnum) {
    PostPaidServerDataVolumeVolumetypeEnum["SATA"] = "SATA";
    PostPaidServerDataVolumeVolumetypeEnum["SAS"] = "SAS";
    PostPaidServerDataVolumeVolumetypeEnum["SSD"] = "SSD";
    PostPaidServerDataVolumeVolumetypeEnum["GPSSD"] = "GPSSD";
    PostPaidServerDataVolumeVolumetypeEnum["CO_P1"] = "co-p1";
    PostPaidServerDataVolumeVolumetypeEnum["UH_L1"] = "uh-l1";
})(PostPaidServerDataVolumeVolumetypeEnum = exports.PostPaidServerDataVolumeVolumetypeEnum || (exports.PostPaidServerDataVolumeVolumetypeEnum = {}));
/**
    * @export
    * @enum {string}
    */
var PostPaidServerDataVolumeClusterTypeEnum;
(function (PostPaidServerDataVolumeClusterTypeEnum) {
    PostPaidServerDataVolumeClusterTypeEnum["DSS"] = "DSS";
})(PostPaidServerDataVolumeClusterTypeEnum = exports.PostPaidServerDataVolumeClusterTypeEnum || (exports.PostPaidServerDataVolumeClusterTypeEnum = {}));
