// @huaweicloud/huaweicloud-sdk-ecs/v2/model/PostPaidServerSchedulerHints.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PostPaidServerSchedulerHints = void 0;
var PostPaidServerSchedulerHints = /** @class */ (function () {
    function PostPaidServerSchedulerHints() {
    }
    PostPaidServerSchedulerHints.prototype.withGroup = function (group) {
        this['group'] = group;
        return this;
    };
    PostPaidServerSchedulerHints.prototype.withDedicatedHostId = function (dedicatedHostId) {
        this['dedicated_host_id'] = dedicatedHostId;
        return this;
    };
    Object.defineProperty(PostPaidServerSchedulerHints.prototype, "dedicatedHostId", {
        get: function () {
            return this['dedicated_host_id'];
        },
        set: function (dedicatedHostId) {
            this['dedicated_host_id'] = dedicatedHostId;
        },
        enumerable: false,
        configurable: true
    });
    PostPaidServerSchedulerHints.prototype.withTenancy = function (tenancy) {
        this['tenancy'] = tenancy;
        return this;
    };
    return PostPaidServerSchedulerHints;
}());
exports.PostPaidServerSchedulerHints = PostPaidServerSchedulerHints;
