// @huaweicloud/huaweicloud-sdk-ecs/v2/model/PostPaidServerRootVolume.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PostPaidServerRootVolumeClusterTypeEnum = exports.PostPaidServerRootVolumeVolumetypeEnum = exports.PostPaidServerRootVolume = void 0;
var PostPaidServerRootVolume = /** @class */ (function () {
    function PostPaidServerRootVolume(volumetype) {
        this['volumetype'] = volumetype;
    }
    PostPaidServerRootVolume.prototype.withVolumetype = function (volumetype) {
        this['volumetype'] = volumetype;
        return this;
    };
    PostPaidServerRootVolume.prototype.withSize = function (size) {
        this['size'] = size;
        return this;
    };
    PostPaidServerRootVolume.prototype.withHwPassthrough = function (hwPassthrough) {
        this['hw:passthrough'] = hwPassthrough;
        return this;
    };
    Object.defineProperty(PostPaidServerRootVolume.prototype, "hwPassthrough", {
        get: function () {
            return this['hw:passthrough'];
        },
        set: function (hwPassthrough) {
            this['hw:passthrough'] = hwPassthrough;
        },
        enumerable: false,
        configurable: true
    });
    PostPaidServerRootVolume.prototype.withClusterType = function (clusterType) {
        this['cluster_type'] = clusterType;
        return this;
    };
    Object.defineProperty(PostPaidServerRootVolume.prototype, "clusterType", {
        get: function () {
            return this['cluster_type'];
        },
        set: function (clusterType) {
            this['cluster_type'] = clusterType;
        },
        enumerable: false,
        configurable: true
    });
    PostPaidServerRootVolume.prototype.withClusterId = function (clusterId) {
        this['cluster_id'] = clusterId;
        return this;
    };
    Object.defineProperty(PostPaidServerRootVolume.prototype, "clusterId", {
        get: function () {
            return this['cluster_id'];
        },
        set: function (clusterId) {
            this['cluster_id'] = clusterId;
        },
        enumerable: false,
        configurable: true
    });
    PostPaidServerRootVolume.prototype.withExtendparam = function (extendparam) {
        this['extendparam'] = extendparam;
        return this;
    };
    return PostPaidServerRootVolume;
}());
exports.PostPaidServerRootVolume = PostPaidServerRootVolume;
/**
    * @export
    * @enum {string}
    */
var PostPaidServerRootVolumeVolumetypeEnum;
(function (PostPaidServerRootVolumeVolumetypeEnum) {
    PostPaidServerRootVolumeVolumetypeEnum["SATA"] = "SATA";
    PostPaidServerRootVolumeVolumetypeEnum["SAS"] = "SAS";
    PostPaidServerRootVolumeVolumetypeEnum["SSD"] = "SSD";
    PostPaidServerRootVolumeVolumetypeEnum["GPSSD"] = "GPSSD";
    PostPaidServerRootVolumeVolumetypeEnum["CO_P1"] = "co-p1";
    PostPaidServerRootVolumeVolumetypeEnum["UH_L1"] = "uh-l1";
})(PostPaidServerRootVolumeVolumetypeEnum = exports.PostPaidServerRootVolumeVolumetypeEnum || (exports.PostPaidServerRootVolumeVolumetypeEnum = {}));
/**
    * @export
    * @enum {string}
    */
var PostPaidServerRootVolumeClusterTypeEnum;
(function (PostPaidServerRootVolumeClusterTypeEnum) {
    PostPaidServerRootVolumeClusterTypeEnum["DSS"] = "DSS";
})(PostPaidServerRootVolumeClusterTypeEnum = exports.PostPaidServerRootVolumeClusterTypeEnum || (exports.PostPaidServerRootVolumeClusterTypeEnum = {}));
