// @huaweicloud/huaweicloud-sdk-ecs/v2/model/PostPaidServerExtendParam.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PostPaidServerExtendParamInterruptionPolicyEnum = exports.PostPaidServerExtendParam = void 0;
var PostPaidServerExtendParam = /** @class */ (function () {
    function PostPaidServerExtendParam() {
    }
    PostPaidServerExtendParam.prototype.withChargingMode = function (chargingMode) {
        this['chargingMode'] = chargingMode;
        return this;
    };
    PostPaidServerExtendParam.prototype.withRegionID = function (regionID) {
        this['regionID'] = regionID;
        return this;
    };
    PostPaidServerExtendParam.prototype.withSupportAutoRecovery = function (supportAutoRecovery) {
        this['support_auto_recovery'] = supportAutoRecovery;
        return this;
    };
    Object.defineProperty(PostPaidServerExtendParam.prototype, "supportAutoRecovery", {
        get: function () {
            return this['support_auto_recovery'];
        },
        set: function (supportAutoRecovery) {
            this['support_auto_recovery'] = supportAutoRecovery;
        },
        enumerable: false,
        configurable: true
    });
    PostPaidServerExtendParam.prototype.withEnterpriseProjectId = function (enterpriseProjectId) {
        this['enterprise_project_id'] = enterpriseProjectId;
        return this;
    };
    Object.defineProperty(PostPaidServerExtendParam.prototype, "enterpriseProjectId", {
        get: function () {
            return this['enterprise_project_id'];
        },
        set: function (enterpriseProjectId) {
            this['enterprise_project_id'] = enterpriseProjectId;
        },
        enumerable: false,
        configurable: true
    });
    PostPaidServerExtendParam.prototype.withMarketType = function (marketType) {
        this['marketType'] = marketType;
        return this;
    };
    PostPaidServerExtendParam.prototype.withSpotPrice = function (spotPrice) {
        this['spotPrice'] = spotPrice;
        return this;
    };
    PostPaidServerExtendParam.prototype.withDiskPrior = function (diskPrior) {
        this['diskPrior'] = diskPrior;
        return this;
    };
    PostPaidServerExtendParam.prototype.withSpotDurationHours = function (spotDurationHours) {
        this['spot_duration_hours'] = spotDurationHours;
        return this;
    };
    Object.defineProperty(PostPaidServerExtendParam.prototype, "spotDurationHours", {
        get: function () {
            return this['spot_duration_hours'];
        },
        set: function (spotDurationHours) {
            this['spot_duration_hours'] = spotDurationHours;
        },
        enumerable: false,
        configurable: true
    });
    PostPaidServerExtendParam.prototype.withInterruptionPolicy = function (interruptionPolicy) {
        this['interruption_policy'] = interruptionPolicy;
        return this;
    };
    Object.defineProperty(PostPaidServerExtendParam.prototype, "interruptionPolicy", {
        get: function () {
            return this['interruption_policy'];
        },
        set: function (interruptionPolicy) {
            this['interruption_policy'] = interruptionPolicy;
        },
        enumerable: false,
        configurable: true
    });
    PostPaidServerExtendParam.prototype.withSpotDurationCount = function (spotDurationCount) {
        this['spot_duration_count'] = spotDurationCount;
        return this;
    };
    Object.defineProperty(PostPaidServerExtendParam.prototype, "spotDurationCount", {
        get: function () {
            return this['spot_duration_count'];
        },
        set: function (spotDurationCount) {
            this['spot_duration_count'] = spotDurationCount;
        },
        enumerable: false,
        configurable: true
    });
    PostPaidServerExtendParam.prototype.withCbCsbsBackup = function (cbCsbsBackup) {
        this['CB_CSBS_BACKUP'] = cbCsbsBackup;
        return this;
    };
    Object.defineProperty(PostPaidServerExtendParam.prototype, "cbCsbsBackup", {
        get: function () {
            return this['CB_CSBS_BACKUP'];
        },
        set: function (cbCsbsBackup) {
            this['CB_CSBS_BACKUP'] = cbCsbsBackup;
        },
        enumerable: false,
        configurable: true
    });
    return PostPaidServerExtendParam;
}());
exports.PostPaidServerExtendParam = PostPaidServerExtendParam;
/**
    * @export
    * @enum {string}
    */
var PostPaidServerExtendParamInterruptionPolicyEnum;
(function (PostPaidServerExtendParamInterruptionPolicyEnum) {
    PostPaidServerExtendParamInterruptionPolicyEnum["IMMEDIATE"] = "immediate";
})(PostPaidServerExtendParamInterruptionPolicyEnum = exports.PostPaidServerExtendParamInterruptionPolicyEnum || (exports.PostPaidServerExtendParamInterruptionPolicyEnum = {}));
