// @huaweicloud/huaweicloud-sdk-ecs/v2/model/PostPaidServerEipExtendParam.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PostPaidServerEipExtendParamChargingModeEnum = exports.PostPaidServerEipExtendParam = void 0;
var PostPaidServerEipExtendParam = /** @class */ (function () {
    function PostPaidServerEipExtendParam() {
    }
    PostPaidServerEipExtendParam.prototype.withChargingMode = function (chargingMode) {
        this['chargingMode'] = chargingMode;
        return this;
    };
    return PostPaidServerEipExtendParam;
}());
exports.PostPaidServerEipExtendParam = PostPaidServerEipExtendParam;
/**
    * @export
    * @enum {string}
    */
var PostPaidServerEipExtendParamChargingModeEnum;
(function (PostPaidServerEipExtendParamChargingModeEnum) {
    PostPaidServerEipExtendParamChargingModeEnum["PREPAID"] = "prePaid";
    PostPaidServerEipExtendParamChargingModeEnum["POSTPAID"] = "postPaid";
})(PostPaidServerEipExtendParamChargingModeEnum = exports.PostPaidServerEipExtendParamChargingModeEnum || (exports.PostPaidServerEipExtendParamChargingModeEnum = {}));
