// @huaweicloud/huaweicloud-sdk-ecs/v2/model/PostPaidServerEipBandwidth.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PostPaidServerEipBandwidthSharetypeEnum = exports.PostPaidServerEipBandwidth = void 0;
var PostPaidServerEipBandwidth = /** @class */ (function () {
    function PostPaidServerEipBandwidth(sharetype) {
        this['sharetype'] = sharetype;
    }
    PostPaidServerEipBandwidth.prototype.withSize = function (size) {
        this['size'] = size;
        return this;
    };
    PostPaidServerEipBandwidth.prototype.withSharetype = function (sharetype) {
        this['sharetype'] = sharetype;
        return this;
    };
    PostPaidServerEipBandwidth.prototype.withChargemode = function (chargemode) {
        this['chargemode'] = chargemode;
        return this;
    };
    PostPaidServerEipBandwidth.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    return PostPaidServerEipBandwidth;
}());
exports.PostPaidServerEipBandwidth = PostPaidServerEipBandwidth;
/**
    * @export
    * @enum {string}
    */
var PostPaidServerEipBandwidthSharetypeEnum;
(function (PostPaidServerEipBandwidthSharetypeEnum) {
    PostPaidServerEipBandwidthSharetypeEnum["PER"] = "PER";
    PostPaidServerEipBandwidthSharetypeEnum["WHOLE"] = "WHOLE";
})(PostPaidServerEipBandwidthSharetypeEnum = exports.PostPaidServerEipBandwidthSharetypeEnum || (exports.PostPaidServerEipBandwidthSharetypeEnum = {}));
