// @huaweicloud/huaweicloud-sdk-ecs/v2/model/PrePaidServerEipExtendParam.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrePaidServerEipExtendParamChargingModeEnum = exports.PrePaidServerEipExtendParam = void 0;
var PrePaidServerEipExtendParam = /** @class */ (function () {
    function PrePaidServerEipExtendParam() {
    }
    PrePaidServerEipExtendParam.prototype.withChargingMode = function (chargingMode) {
        this['chargingMode'] = chargingMode;
        return this;
    };
    return PrePaidServerEipExtendParam;
}());
exports.PrePaidServerEipExtendParam = PrePaidServerEipExtendParam;
/**
    * @export
    * @enum {string}
    */
var PrePaidServerEipExtendParamChargingModeEnum;
(function (PrePaidServerEipExtendParamChargingModeEnum) {
    PrePaidServerEipExtendParamChargingModeEnum["PREPAID"] = "prePaid";
    PrePaidServerEipExtendParamChargingModeEnum["POSTPAID"] = "postPaid";
})(PrePaidServerEipExtendParamChargingModeEnum = exports.PrePaidServerEipExtendParamChargingModeEnum || (exports.PrePaidServerEipExtendParamChargingModeEnum = {}));
