// @huaweicloud/huaweicloud-sdk-ecs/v2/model/PrePaidServerSchedulerHints.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrePaidServerSchedulerHintsTenancyEnum = exports.PrePaidServerSchedulerHints = void 0;
var PrePaidServerSchedulerHints = /** @class */ (function () {
    function PrePaidServerSchedulerHints() {
    }
    PrePaidServerSchedulerHints.prototype.withGroup = function (group) {
        this['group'] = group;
        return this;
    };
    PrePaidServerSchedulerHints.prototype.withTenancy = function (tenancy) {
        this['tenancy'] = tenancy;
        return this;
    };
    PrePaidServerSchedulerHints.prototype.withDedicatedHostId = function (dedicatedHostId) {
        this['dedicated_host_id'] = dedicatedHostId;
        return this;
    };
    Object.defineProperty(PrePaidServerSchedulerHints.prototype, "dedicatedHostId", {
        get: function () {
            return this['dedicated_host_id'];
        },
        set: function (dedicatedHostId) {
            this['dedicated_host_id'] = dedicatedHostId;
        },
        enumerable: false,
        configurable: true
    });
    return PrePaidServerSchedulerHints;
}());
exports.PrePaidServerSchedulerHints = PrePaidServerSchedulerHints;
/**
    * @export
    * @enum {string}
    */
var PrePaidServerSchedulerHintsTenancyEnum;
(function (PrePaidServerSchedulerHintsTenancyEnum) {
    PrePaidServerSchedulerHintsTenancyEnum["SHARED"] = "shared";
    PrePaidServerSchedulerHintsTenancyEnum["DEDICATED"] = "dedicated";
})(PrePaidServerSchedulerHintsTenancyEnum = exports.PrePaidServerSchedulerHintsTenancyEnum || (exports.PrePaidServerSchedulerHintsTenancyEnum = {}));
