// @huaweicloud/huaweicloud-sdk-ecs/v2/model/PrePaidServerEipBandwidth.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrePaidServerEipBandwidthSharetypeEnum = exports.PrePaidServerEipBandwidth = void 0;
var PrePaidServerEipBandwidth = /** @class */ (function () {
    function PrePaidServerEipBandwidth(sharetype) {
        this['sharetype'] = sharetype;
    }
    PrePaidServerEipBandwidth.prototype.withSize = function (size) {
        this['size'] = size;
        return this;
    };
    PrePaidServerEipBandwidth.prototype.withSharetype = function (sharetype) {
        this['sharetype'] = sharetype;
        return this;
    };
    PrePaidServerEipBandwidth.prototype.withChargemode = function (chargemode) {
        this['chargemode'] = chargemode;
        return this;
    };
    PrePaidServerEipBandwidth.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    return PrePaidServerEipBandwidth;
}());
exports.PrePaidServerEipBandwidth = PrePaidServerEipBandwidth;
/**
    * @export
    * @enum {string}
    */
var PrePaidServerEipBandwidthSharetypeEnum;
(function (PrePaidServerEipBandwidthSharetypeEnum) {
    PrePaidServerEipBandwidthSharetypeEnum["PER"] = "PER";
    PrePaidServerEipBandwidthSharetypeEnum["WHOLE"] = "WHOLE";
})(PrePaidServerEipBandwidthSharetypeEnum = exports.PrePaidServerEipBandwidthSharetypeEnum || (exports.PrePaidServerEipBandwidthSharetypeEnum = {}));
