// @huaweicloud/huaweicloud-sdk-ecs/v2/model/PrePaidServerNic.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrePaidServerNic = void 0;
var PrePaidServerNic = /** @class */ (function () {
    function PrePaidServerNic(subnetId) {
        this['subnet_id'] = subnetId;
    }
    PrePaidServerNic.prototype.withSubnetId = function (subnetId) {
        this['subnet_id'] = subnetId;
        return this;
    };
    Object.defineProperty(PrePaidServerNic.prototype, "subnetId", {
        get: function () {
            return this['subnet_id'];
        },
        set: function (subnetId) {
            this['subnet_id'] = subnetId;
        },
        enumerable: false,
        configurable: true
    });
    PrePaidServerNic.prototype.withIpAddress = function (ipAddress) {
        this['ip_address'] = ipAddress;
        return this;
    };
    Object.defineProperty(PrePaidServerNic.prototype, "ipAddress", {
        get: function () {
            return this['ip_address'];
        },
        set: function (ipAddress) {
            this['ip_address'] = ipAddress;
        },
        enumerable: false,
        configurable: true
    });
    PrePaidServerNic.prototype.withIpv6Enable = function (ipv6Enable) {
        this['ipv6_enable'] = ipv6Enable;
        return this;
    };
    Object.defineProperty(PrePaidServerNic.prototype, "ipv6Enable", {
        get: function () {
            return this['ipv6_enable'];
        },
        set: function (ipv6Enable) {
            this['ipv6_enable'] = ipv6Enable;
        },
        enumerable: false,
        configurable: true
    });
    PrePaidServerNic.prototype.withIpv6Bandwidth = function (ipv6Bandwidth) {
        this['ipv6_bandwidth'] = ipv6Bandwidth;
        return this;
    };
    Object.defineProperty(PrePaidServerNic.prototype, "ipv6Bandwidth", {
        get: function () {
            return this['ipv6_bandwidth'];
        },
        set: function (ipv6Bandwidth) {
            this['ipv6_bandwidth'] = ipv6Bandwidth;
        },
        enumerable: false,
        configurable: true
    });
    return PrePaidServerNic;
}());
exports.PrePaidServerNic = PrePaidServerNic;
