// @huaweicloud/huaweicloud-sdk-ecs/v2/model/PrePaidServerExtendParam.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrePaidServerExtendParamInterruptionPolicyEnum = exports.PrePaidServerExtendParamIsAutoPayEnum = exports.PrePaidServerExtendParamIsAutoRenewEnum = exports.PrePaidServerExtendParamPeriodTypeEnum = exports.PrePaidServerExtendParamChargingModeEnum = exports.PrePaidServerExtendParam = void 0;
var PrePaidServerExtendParam = /** @class */ (function () {
    function PrePaidServerExtendParam() {
    }
    PrePaidServerExtendParam.prototype.withChargingMode = function (chargingMode) {
        this['chargingMode'] = chargingMode;
        return this;
    };
    PrePaidServerExtendParam.prototype.withRegionID = function (regionID) {
        this['regionID'] = regionID;
        return this;
    };
    PrePaidServerExtendParam.prototype.withPeriodType = function (periodType) {
        this['periodType'] = periodType;
        return this;
    };
    PrePaidServerExtendParam.prototype.withPeriodNum = function (periodNum) {
        this['periodNum'] = periodNum;
        return this;
    };
    PrePaidServerExtendParam.prototype.withIsAutoRenew = function (isAutoRenew) {
        this['isAutoRenew'] = isAutoRenew;
        return this;
    };
    PrePaidServerExtendParam.prototype.withIsAutoPay = function (isAutoPay) {
        this['isAutoPay'] = isAutoPay;
        return this;
    };
    PrePaidServerExtendParam.prototype.withEnterpriseProjectId = function (enterpriseProjectId) {
        this['enterprise_project_id'] = enterpriseProjectId;
        return this;
    };
    Object.defineProperty(PrePaidServerExtendParam.prototype, "enterpriseProjectId", {
        get: function () {
            return this['enterprise_project_id'];
        },
        set: function (enterpriseProjectId) {
            this['enterprise_project_id'] = enterpriseProjectId;
        },
        enumerable: false,
        configurable: true
    });
    PrePaidServerExtendParam.prototype.withSupportAutoRecovery = function (supportAutoRecovery) {
        this['support_auto_recovery'] = supportAutoRecovery;
        return this;
    };
    Object.defineProperty(PrePaidServerExtendParam.prototype, "supportAutoRecovery", {
        get: function () {
            return this['support_auto_recovery'];
        },
        set: function (supportAutoRecovery) {
            this['support_auto_recovery'] = supportAutoRecovery;
        },
        enumerable: false,
        configurable: true
    });
    PrePaidServerExtendParam.prototype.withMarketType = function (marketType) {
        this['marketType'] = marketType;
        return this;
    };
    PrePaidServerExtendParam.prototype.withSpotPrice = function (spotPrice) {
        this['spotPrice'] = spotPrice;
        return this;
    };
    PrePaidServerExtendParam.prototype.withDiskPrior = function (diskPrior) {
        this['diskPrior'] = diskPrior;
        return this;
    };
    PrePaidServerExtendParam.prototype.withSpotDurationHours = function (spotDurationHours) {
        this['spot_duration_hours'] = spotDurationHours;
        return this;
    };
    Object.defineProperty(PrePaidServerExtendParam.prototype, "spotDurationHours", {
        get: function () {
            return this['spot_duration_hours'];
        },
        set: function (spotDurationHours) {
            this['spot_duration_hours'] = spotDurationHours;
        },
        enumerable: false,
        configurable: true
    });
    PrePaidServerExtendParam.prototype.withInterruptionPolicy = function (interruptionPolicy) {
        this['interruption_policy'] = interruptionPolicy;
        return this;
    };
    Object.defineProperty(PrePaidServerExtendParam.prototype, "interruptionPolicy", {
        get: function () {
            return this['interruption_policy'];
        },
        set: function (interruptionPolicy) {
            this['interruption_policy'] = interruptionPolicy;
        },
        enumerable: false,
        configurable: true
    });
    PrePaidServerExtendParam.prototype.withSpotDurationCount = function (spotDurationCount) {
        this['spot_duration_count'] = spotDurationCount;
        return this;
    };
    Object.defineProperty(PrePaidServerExtendParam.prototype, "spotDurationCount", {
        get: function () {
            return this['spot_duration_count'];
        },
        set: function (spotDurationCount) {
            this['spot_duration_count'] = spotDurationCount;
        },
        enumerable: false,
        configurable: true
    });
    return PrePaidServerExtendParam;
}());
exports.PrePaidServerExtendParam = PrePaidServerExtendParam;
/**
    * @export
    * @enum {string}
    */
var PrePaidServerExtendParamChargingModeEnum;
(function (PrePaidServerExtendParamChargingModeEnum) {
    PrePaidServerExtendParamChargingModeEnum["PREPAID"] = "prePaid";
    PrePaidServerExtendParamChargingModeEnum["POSTPAID"] = "postPaid";
})(PrePaidServerExtendParamChargingModeEnum = exports.PrePaidServerExtendParamChargingModeEnum || (exports.PrePaidServerExtendParamChargingModeEnum = {}));
/**
    * @export
    * @enum {string}
    */
var PrePaidServerExtendParamPeriodTypeEnum;
(function (PrePaidServerExtendParamPeriodTypeEnum) {
    PrePaidServerExtendParamPeriodTypeEnum["MONTH"] = "month";
    PrePaidServerExtendParamPeriodTypeEnum["YEAR"] = "year";
})(PrePaidServerExtendParamPeriodTypeEnum = exports.PrePaidServerExtendParamPeriodTypeEnum || (exports.PrePaidServerExtendParamPeriodTypeEnum = {}));
/**
    * @export
    * @enum {string}
    */
var PrePaidServerExtendParamIsAutoRenewEnum;
(function (PrePaidServerExtendParamIsAutoRenewEnum) {
    PrePaidServerExtendParamIsAutoRenewEnum["TRUE"] = "true";
    PrePaidServerExtendParamIsAutoRenewEnum["FALSE"] = "false";
})(PrePaidServerExtendParamIsAutoRenewEnum = exports.PrePaidServerExtendParamIsAutoRenewEnum || (exports.PrePaidServerExtendParamIsAutoRenewEnum = {}));
/**
    * @export
    * @enum {string}
    */
var PrePaidServerExtendParamIsAutoPayEnum;
(function (PrePaidServerExtendParamIsAutoPayEnum) {
    PrePaidServerExtendParamIsAutoPayEnum["TRUE"] = "true";
    PrePaidServerExtendParamIsAutoPayEnum["FALSE"] = "false";
})(PrePaidServerExtendParamIsAutoPayEnum = exports.PrePaidServerExtendParamIsAutoPayEnum || (exports.PrePaidServerExtendParamIsAutoPayEnum = {}));
/**
    * @export
    * @enum {string}
    */
var PrePaidServerExtendParamInterruptionPolicyEnum;
(function (PrePaidServerExtendParamInterruptionPolicyEnum) {
    PrePaidServerExtendParamInterruptionPolicyEnum["IMMEDIATE"] = "immediate";
})(PrePaidServerExtendParamInterruptionPolicyEnum = exports.PrePaidServerExtendParamInterruptionPolicyEnum || (exports.PrePaidServerExtendParamInterruptionPolicyEnum = {}));
