// @huaweicloud/huaweicloud-sdk-ecs/v2/model/PrePaidServerDataVolume.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrePaidServerDataVolumeClusterTypeEnum = exports.PrePaidServerDataVolumeVolumetypeEnum = exports.PrePaidServerDataVolume = void 0;
var PrePaidServerDataVolume = /** @class */ (function () {
    function PrePaidServerDataVolume(volumetype, size) {
        this['volumetype'] = volumetype;
        this['size'] = size;
    }
    PrePaidServerDataVolume.prototype.withVolumetype = function (volumetype) {
        this['volumetype'] = volumetype;
        return this;
    };
    PrePaidServerDataVolume.prototype.withSize = function (size) {
        this['size'] = size;
        return this;
    };
    PrePaidServerDataVolume.prototype.withShareable = function (shareable) {
        this['shareable'] = shareable;
        return this;
    };
    PrePaidServerDataVolume.prototype.withMultiattach = function (multiattach) {
        this['multiattach'] = multiattach;
        return this;
    };
    PrePaidServerDataVolume.prototype.withHwPassthrough = function (hwPassthrough) {
        this['hw:passthrough'] = hwPassthrough;
        return this;
    };
    Object.defineProperty(PrePaidServerDataVolume.prototype, "hwPassthrough", {
        get: function () {
            return this['hw:passthrough'];
        },
        set: function (hwPassthrough) {
            this['hw:passthrough'] = hwPassthrough;
        },
        enumerable: false,
        configurable: true
    });
    PrePaidServerDataVolume.prototype.withExtendparam = function (extendparam) {
        this['extendparam'] = extendparam;
        return this;
    };
    PrePaidServerDataVolume.prototype.withClusterType = function (clusterType) {
        this['cluster_type'] = clusterType;
        return this;
    };
    Object.defineProperty(PrePaidServerDataVolume.prototype, "clusterType", {
        get: function () {
            return this['cluster_type'];
        },
        set: function (clusterType) {
            this['cluster_type'] = clusterType;
        },
        enumerable: false,
        configurable: true
    });
    PrePaidServerDataVolume.prototype.withClusterId = function (clusterId) {
        this['cluster_id'] = clusterId;
        return this;
    };
    Object.defineProperty(PrePaidServerDataVolume.prototype, "clusterId", {
        get: function () {
            return this['cluster_id'];
        },
        set: function (clusterId) {
            this['cluster_id'] = clusterId;
        },
        enumerable: false,
        configurable: true
    });
    PrePaidServerDataVolume.prototype.withMetadata = function (metadata) {
        this['metadata'] = metadata;
        return this;
    };
    PrePaidServerDataVolume.prototype.withDataImageId = function (dataImageId) {
        this['data_image_id'] = dataImageId;
        return this;
    };
    Object.defineProperty(PrePaidServerDataVolume.prototype, "dataImageId", {
        get: function () {
            return this['data_image_id'];
        },
        set: function (dataImageId) {
            this['data_image_id'] = dataImageId;
        },
        enumerable: false,
        configurable: true
    });
    return PrePaidServerDataVolume;
}());
exports.PrePaidServerDataVolume = PrePaidServerDataVolume;
/**
    * @export
    * @enum {string}
    */
var PrePaidServerDataVolumeVolumetypeEnum;
(function (PrePaidServerDataVolumeVolumetypeEnum) {
    PrePaidServerDataVolumeVolumetypeEnum["SATA"] = "SATA";
    PrePaidServerDataVolumeVolumetypeEnum["SAS"] = "SAS";
    PrePaidServerDataVolumeVolumetypeEnum["SSD"] = "SSD";
    PrePaidServerDataVolumeVolumetypeEnum["GPSSD"] = "GPSSD";
    PrePaidServerDataVolumeVolumetypeEnum["CO_P1"] = "co-p1";
    PrePaidServerDataVolumeVolumetypeEnum["UH_L1"] = "uh-l1";
})(PrePaidServerDataVolumeVolumetypeEnum = exports.PrePaidServerDataVolumeVolumetypeEnum || (exports.PrePaidServerDataVolumeVolumetypeEnum = {}));
/**
    * @export
    * @enum {string}
    */
var PrePaidServerDataVolumeClusterTypeEnum;
(function (PrePaidServerDataVolumeClusterTypeEnum) {
    PrePaidServerDataVolumeClusterTypeEnum["DSS"] = "DSS";
})(PrePaidServerDataVolumeClusterTypeEnum = exports.PrePaidServerDataVolumeClusterTypeEnum || (exports.PrePaidServerDataVolumeClusterTypeEnum = {}));
