// @huaweicloud/huaweicloud-sdk-ecs/v2/model/ReinstallServerWithoutCloudInitOption.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReinstallServerWithoutCloudInitOption = void 0;
var ReinstallServerWithoutCloudInitOption = /** @class */ (function () {
    function ReinstallServerWithoutCloudInitOption() {
    }
    ReinstallServerWithoutCloudInitOption.prototype.withAdminpass = function (adminpass) {
        this['adminpass'] = adminpass;
        return this;
    };
    ReinstallServerWithoutCloudInitOption.prototype.withKeyname = function (keyname) {
        this['keyname'] = keyname;
        return this;
    };
    ReinstallServerWithoutCloudInitOption.prototype.withUserid = function (userid) {
        this['userid'] = userid;
        return this;
    };
    ReinstallServerWithoutCloudInitOption.prototype.withMode = function (mode) {
        this['mode'] = mode;
        return this;
    };
    return ReinstallServerWithoutCloudInitOption;
}());
exports.ReinstallServerWithoutCloudInitOption = ReinstallServerWithoutCloudInitOption;
