// @huaweicloud/huaweicloud-sdk-ecs/v2/model/ReinstallServerWithCloudInitOption.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReinstallServerWithCloudInitOption = void 0;
var ReinstallServerWithCloudInitOption = /** @class */ (function () {
    function ReinstallServerWithCloudInitOption() {
    }
    ReinstallServerWithCloudInitOption.prototype.withAdminpass = function (adminpass) {
        this['adminpass'] = adminpass;
        return this;
    };
    ReinstallServerWithCloudInitOption.prototype.withKeyname = function (keyname) {
        this['keyname'] = keyname;
        return this;
    };
    ReinstallServerWithCloudInitOption.prototype.withUserid = function (userid) {
        this['userid'] = userid;
        return this;
    };
    ReinstallServerWithCloudInitOption.prototype.withMetadata = function (metadata) {
        this['metadata'] = metadata;
        return this;
    };
    ReinstallServerWithCloudInitOption.prototype.withMode = function (mode) {
        this['mode'] = mode;
        return this;
    };
    return ReinstallServerWithCloudInitOption;
}());
exports.ReinstallServerWithCloudInitOption = ReinstallServerWithCloudInitOption;
