// @huaweicloud/huaweicloud-sdk-ecs/v2/model/ResizePrePaidServerOption.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResizePrePaidServerOption = void 0;
var ResizePrePaidServerOption = /** @class */ (function () {
    function ResizePrePaidServerOption(flavorRef) {
        this['flavorRef'] = flavorRef;
    }
    ResizePrePaidServerOption.prototype.withFlavorRef = function (flavorRef) {
        this['flavorRef'] = flavorRef;
        return this;
    };
    ResizePrePaidServerOption.prototype.withDedicatedHostId = function (dedicatedHostId) {
        this['dedicated_host_id'] = dedicatedHostId;
        return this;
    };
    Object.defineProperty(ResizePrePaidServerOption.prototype, "dedicatedHostId", {
        get: function () {
            return this['dedicated_host_id'];
        },
        set: function (dedicatedHostId) {
            this['dedicated_host_id'] = dedicatedHostId;
        },
        enumerable: false,
        configurable: true
    });
    ResizePrePaidServerOption.prototype.withExtendparam = function (extendparam) {
        this['extendparam'] = extendparam;
        return this;
    };
    ResizePrePaidServerOption.prototype.withMode = function (mode) {
        this['mode'] = mode;
        return this;
    };
    return ResizePrePaidServerOption;
}());
exports.ResizePrePaidServerOption = ResizePrePaidServerOption;
