// @huaweicloud/huaweicloud-sdk-ecs/v2/model/ServerDetail.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServerDetail = void 0;
var ServerDetail = /** @class */ (function () {
    function ServerDetail(status, updated, autoTerminateTime, hostId, oSEXTSRVATTRHost, addresses, keyName, image, oSEXTSTSTaskState, oSEXTSTSVmState, oSEXTSRVATTRInstanceName, oSEXTSRVATTRHypervisorHostname, flavor, id, securityGroups, oSEXTAZAvailabilityZone, userId, name, created, tenantId, accessIPv4, accessIPv6, oSEXTSTSPowerState, configDrive, metadata, oSSRVUSGLaunchedAt, oSSRVUSGTerminatedAt, osExtendedVolumesVolumesAttached, hostStatus, oSEXTSRVATTRHostname, oSEXTSRVATTRLaunchIndex, oSEXTSRVATTRKernelId, oSEXTSRVATTRRamdiskId, oSEXTSRVATTRRootDeviceName, locked) {
        this['status'] = status;
        this['updated'] = updated;
        this['auto_terminate_time'] = autoTerminateTime;
        this['hostId'] = hostId;
        this['OS-EXT-SRV-ATTR:host'] = oSEXTSRVATTRHost;
        this['addresses'] = addresses;
        this['key_name'] = keyName;
        this['image'] = image;
        this['OS-EXT-STS:task_state'] = oSEXTSTSTaskState;
        this['OS-EXT-STS:vm_state'] = oSEXTSTSVmState;
        this['OS-EXT-SRV-ATTR:instance_name'] = oSEXTSRVATTRInstanceName;
        this['OS-EXT-SRV-ATTR:hypervisor_hostname'] = oSEXTSRVATTRHypervisorHostname;
        this['flavor'] = flavor;
        this['id'] = id;
        this['security_groups'] = securityGroups;
        this['OS-EXT-AZ:availability_zone'] = oSEXTAZAvailabilityZone;
        this['user_id'] = userId;
        this['name'] = name;
        this['created'] = created;
        this['tenant_id'] = tenantId;
        this['accessIPv4'] = accessIPv4;
        this['accessIPv6'] = accessIPv6;
        this['OS-EXT-STS:power_state'] = oSEXTSTSPowerState;
        this['config_drive'] = configDrive;
        this['metadata'] = metadata;
        this['OS-SRV-USG:launched_at'] = oSSRVUSGLaunchedAt;
        this['OS-SRV-USG:terminated_at'] = oSSRVUSGTerminatedAt;
        this['os-extended-volumes:volumes_attached'] = osExtendedVolumesVolumesAttached;
        this['host_status'] = hostStatus;
        this['OS-EXT-SRV-ATTR:hostname'] = oSEXTSRVATTRHostname;
        this['OS-EXT-SRV-ATTR:launch_index'] = oSEXTSRVATTRLaunchIndex;
        this['OS-EXT-SRV-ATTR:kernel_id'] = oSEXTSRVATTRKernelId;
        this['OS-EXT-SRV-ATTR:ramdisk_id'] = oSEXTSRVATTRRamdiskId;
        this['OS-EXT-SRV-ATTR:root_device_name'] = oSEXTSRVATTRRootDeviceName;
        this['locked'] = locked;
    }
    ServerDetail.prototype.withStatus = function (status) {
        this['status'] = status;
        return this;
    };
    ServerDetail.prototype.withUpdated = function (updated) {
        this['updated'] = updated;
        return this;
    };
    ServerDetail.prototype.withAutoTerminateTime = function (autoTerminateTime) {
        this['auto_terminate_time'] = autoTerminateTime;
        return this;
    };
    Object.defineProperty(ServerDetail.prototype, "autoTerminateTime", {
        get: function () {
            return this['auto_terminate_time'];
        },
        set: function (autoTerminateTime) {
            this['auto_terminate_time'] = autoTerminateTime;
        },
        enumerable: false,
        configurable: true
    });
    ServerDetail.prototype.withHostId = function (hostId) {
        this['hostId'] = hostId;
        return this;
    };
    ServerDetail.prototype.withOSEXTSRVATTRHost = function (oSEXTSRVATTRHost) {
        this['OS-EXT-SRV-ATTR:host'] = oSEXTSRVATTRHost;
        return this;
    };
    Object.defineProperty(ServerDetail.prototype, "oSEXTSRVATTRHost", {
        get: function () {
            return this['OS-EXT-SRV-ATTR:host'];
        },
        set: function (oSEXTSRVATTRHost) {
            this['OS-EXT-SRV-ATTR:host'] = oSEXTSRVATTRHost;
        },
        enumerable: false,
        configurable: true
    });
    ServerDetail.prototype.withAddresses = function (addresses) {
        this['addresses'] = addresses;
        return this;
    };
    ServerDetail.prototype.withKeyName = function (keyName) {
        this['key_name'] = keyName;
        return this;
    };
    Object.defineProperty(ServerDetail.prototype, "keyName", {
        get: function () {
            return this['key_name'];
        },
        set: function (keyName) {
            this['key_name'] = keyName;
        },
        enumerable: false,
        configurable: true
    });
    ServerDetail.prototype.withImage = function (image) {
        this['image'] = image;
        return this;
    };
    ServerDetail.prototype.withOSEXTSTSTaskState = function (oSEXTSTSTaskState) {
        this['OS-EXT-STS:task_state'] = oSEXTSTSTaskState;
        return this;
    };
    Object.defineProperty(ServerDetail.prototype, "oSEXTSTSTaskState", {
        get: function () {
            return this['OS-EXT-STS:task_state'];
        },
        set: function (oSEXTSTSTaskState) {
            this['OS-EXT-STS:task_state'] = oSEXTSTSTaskState;
        },
        enumerable: false,
        configurable: true
    });
    ServerDetail.prototype.withOSEXTSTSVmState = function (oSEXTSTSVmState) {
        this['OS-EXT-STS:vm_state'] = oSEXTSTSVmState;
        return this;
    };
    Object.defineProperty(ServerDetail.prototype, "oSEXTSTSVmState", {
        get: function () {
            return this['OS-EXT-STS:vm_state'];
        },
        set: function (oSEXTSTSVmState) {
            this['OS-EXT-STS:vm_state'] = oSEXTSTSVmState;
        },
        enumerable: false,
        configurable: true
    });
    ServerDetail.prototype.withOSEXTSRVATTRInstanceName = function (oSEXTSRVATTRInstanceName) {
        this['OS-EXT-SRV-ATTR:instance_name'] = oSEXTSRVATTRInstanceName;
        return this;
    };
    Object.defineProperty(ServerDetail.prototype, "oSEXTSRVATTRInstanceName", {
        get: function () {
            return this['OS-EXT-SRV-ATTR:instance_name'];
        },
        set: function (oSEXTSRVATTRInstanceName) {
            this['OS-EXT-SRV-ATTR:instance_name'] = oSEXTSRVATTRInstanceName;
        },
        enumerable: false,
        configurable: true
    });
    ServerDetail.prototype.withOSEXTSRVATTRHypervisorHostname = function (oSEXTSRVATTRHypervisorHostname) {
        this['OS-EXT-SRV-ATTR:hypervisor_hostname'] = oSEXTSRVATTRHypervisorHostname;
        return this;
    };
    Object.defineProperty(ServerDetail.prototype, "oSEXTSRVATTRHypervisorHostname", {
        get: function () {
            return this['OS-EXT-SRV-ATTR:hypervisor_hostname'];
        },
        set: function (oSEXTSRVATTRHypervisorHostname) {
            this['OS-EXT-SRV-ATTR:hypervisor_hostname'] = oSEXTSRVATTRHypervisorHostname;
        },
        enumerable: false,
        configurable: true
    });
    ServerDetail.prototype.withFlavor = function (flavor) {
        this['flavor'] = flavor;
        return this;
    };
    ServerDetail.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    ServerDetail.prototype.withSecurityGroups = function (securityGroups) {
        this['security_groups'] = securityGroups;
        return this;
    };
    Object.defineProperty(ServerDetail.prototype, "securityGroups", {
        get: function () {
            return this['security_groups'];
        },
        set: function (securityGroups) {
            this['security_groups'] = securityGroups;
        },
        enumerable: false,
        configurable: true
    });
    ServerDetail.prototype.withOSEXTAZAvailabilityZone = function (oSEXTAZAvailabilityZone) {
        this['OS-EXT-AZ:availability_zone'] = oSEXTAZAvailabilityZone;
        return this;
    };
    Object.defineProperty(ServerDetail.prototype, "oSEXTAZAvailabilityZone", {
        get: function () {
            return this['OS-EXT-AZ:availability_zone'];
        },
        set: function (oSEXTAZAvailabilityZone) {
            this['OS-EXT-AZ:availability_zone'] = oSEXTAZAvailabilityZone;
        },
        enumerable: false,
        configurable: true
    });
    ServerDetail.prototype.withUserId = function (userId) {
        this['user_id'] = userId;
        return this;
    };
    Object.defineProperty(ServerDetail.prototype, "userId", {
        get: function () {
            return this['user_id'];
        },
        set: function (userId) {
            this['user_id'] = userId;
        },
        enumerable: false,
        configurable: true
    });
    ServerDetail.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    ServerDetail.prototype.withCreated = function (created) {
        this['created'] = created;
        return this;
    };
    ServerDetail.prototype.withTenantId = function (tenantId) {
        this['tenant_id'] = tenantId;
        return this;
    };
    Object.defineProperty(ServerDetail.prototype, "tenantId", {
        get: function () {
            return this['tenant_id'];
        },
        set: function (tenantId) {
            this['tenant_id'] = tenantId;
        },
        enumerable: false,
        configurable: true
    });
    ServerDetail.prototype.withOSDCFDiskConfig = function (oSDCFDiskConfig) {
        this['OS-DCF:diskConfig'] = oSDCFDiskConfig;
        return this;
    };
    Object.defineProperty(ServerDetail.prototype, "oSDCFDiskConfig", {
        get: function () {
            return this['OS-DCF:diskConfig'];
        },
        set: function (oSDCFDiskConfig) {
            this['OS-DCF:diskConfig'] = oSDCFDiskConfig;
        },
        enumerable: false,
        configurable: true
    });
    ServerDetail.prototype.withAccessIPv4 = function (accessIPv4) {
        this['accessIPv4'] = accessIPv4;
        return this;
    };
    ServerDetail.prototype.withAccessIPv6 = function (accessIPv6) {
        this['accessIPv6'] = accessIPv6;
        return this;
    };
    ServerDetail.prototype.withFault = function (fault) {
        this['fault'] = fault;
        return this;
    };
    ServerDetail.prototype.withProgress = function (progress) {
        this['progress'] = progress;
        return this;
    };
    ServerDetail.prototype.withOSEXTSTSPowerState = function (oSEXTSTSPowerState) {
        this['OS-EXT-STS:power_state'] = oSEXTSTSPowerState;
        return this;
    };
    Object.defineProperty(ServerDetail.prototype, "oSEXTSTSPowerState", {
        get: function () {
            return this['OS-EXT-STS:power_state'];
        },
        set: function (oSEXTSTSPowerState) {
            this['OS-EXT-STS:power_state'] = oSEXTSTSPowerState;
        },
        enumerable: false,
        configurable: true
    });
    ServerDetail.prototype.withConfigDrive = function (configDrive) {
        this['config_drive'] = configDrive;
        return this;
    };
    Object.defineProperty(ServerDetail.prototype, "configDrive", {
        get: function () {
            return this['config_drive'];
        },
        set: function (configDrive) {
            this['config_drive'] = configDrive;
        },
        enumerable: false,
        configurable: true
    });
    ServerDetail.prototype.withMetadata = function (metadata) {
        this['metadata'] = metadata;
        return this;
    };
    ServerDetail.prototype.withOSSRVUSGLaunchedAt = function (oSSRVUSGLaunchedAt) {
        this['OS-SRV-USG:launched_at'] = oSSRVUSGLaunchedAt;
        return this;
    };
    Object.defineProperty(ServerDetail.prototype, "oSSRVUSGLaunchedAt", {
        get: function () {
            return this['OS-SRV-USG:launched_at'];
        },
        set: function (oSSRVUSGLaunchedAt) {
            this['OS-SRV-USG:launched_at'] = oSSRVUSGLaunchedAt;
        },
        enumerable: false,
        configurable: true
    });
    ServerDetail.prototype.withOSSRVUSGTerminatedAt = function (oSSRVUSGTerminatedAt) {
        this['OS-SRV-USG:terminated_at'] = oSSRVUSGTerminatedAt;
        return this;
    };
    Object.defineProperty(ServerDetail.prototype, "oSSRVUSGTerminatedAt", {
        get: function () {
            return this['OS-SRV-USG:terminated_at'];
        },
        set: function (oSSRVUSGTerminatedAt) {
            this['OS-SRV-USG:terminated_at'] = oSSRVUSGTerminatedAt;
        },
        enumerable: false,
        configurable: true
    });
    ServerDetail.prototype.withOsExtendedVolumesVolumesAttached = function (osExtendedVolumesVolumesAttached) {
        this['os-extended-volumes:volumes_attached'] = osExtendedVolumesVolumesAttached;
        return this;
    };
    Object.defineProperty(ServerDetail.prototype, "osExtendedVolumesVolumesAttached", {
        get: function () {
            return this['os-extended-volumes:volumes_attached'];
        },
        set: function (osExtendedVolumesVolumesAttached) {
            this['os-extended-volumes:volumes_attached'] = osExtendedVolumesVolumesAttached;
        },
        enumerable: false,
        configurable: true
    });
    ServerDetail.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    ServerDetail.prototype.withHostStatus = function (hostStatus) {
        this['host_status'] = hostStatus;
        return this;
    };
    Object.defineProperty(ServerDetail.prototype, "hostStatus", {
        get: function () {
            return this['host_status'];
        },
        set: function (hostStatus) {
            this['host_status'] = hostStatus;
        },
        enumerable: false,
        configurable: true
    });
    ServerDetail.prototype.withOSEXTSRVATTRHostname = function (oSEXTSRVATTRHostname) {
        this['OS-EXT-SRV-ATTR:hostname'] = oSEXTSRVATTRHostname;
        return this;
    };
    Object.defineProperty(ServerDetail.prototype, "oSEXTSRVATTRHostname", {
        get: function () {
            return this['OS-EXT-SRV-ATTR:hostname'];
        },
        set: function (oSEXTSRVATTRHostname) {
            this['OS-EXT-SRV-ATTR:hostname'] = oSEXTSRVATTRHostname;
        },
        enumerable: false,
        configurable: true
    });
    ServerDetail.prototype.withOSEXTSRVATTRReservationId = function (oSEXTSRVATTRReservationId) {
        this['OS-EXT-SRV-ATTR:reservation_id'] = oSEXTSRVATTRReservationId;
        return this;
    };
    Object.defineProperty(ServerDetail.prototype, "oSEXTSRVATTRReservationId", {
        get: function () {
            return this['OS-EXT-SRV-ATTR:reservation_id'];
        },
        set: function (oSEXTSRVATTRReservationId) {
            this['OS-EXT-SRV-ATTR:reservation_id'] = oSEXTSRVATTRReservationId;
        },
        enumerable: false,
        configurable: true
    });
    ServerDetail.prototype.withOSEXTSRVATTRLaunchIndex = function (oSEXTSRVATTRLaunchIndex) {
        this['OS-EXT-SRV-ATTR:launch_index'] = oSEXTSRVATTRLaunchIndex;
        return this;
    };
    Object.defineProperty(ServerDetail.prototype, "oSEXTSRVATTRLaunchIndex", {
        get: function () {
            return this['OS-EXT-SRV-ATTR:launch_index'];
        },
        set: function (oSEXTSRVATTRLaunchIndex) {
            this['OS-EXT-SRV-ATTR:launch_index'] = oSEXTSRVATTRLaunchIndex;
        },
        enumerable: false,
        configurable: true
    });
    ServerDetail.prototype.withOSEXTSRVATTRKernelId = function (oSEXTSRVATTRKernelId) {
        this['OS-EXT-SRV-ATTR:kernel_id'] = oSEXTSRVATTRKernelId;
        return this;
    };
    Object.defineProperty(ServerDetail.prototype, "oSEXTSRVATTRKernelId", {
        get: function () {
            return this['OS-EXT-SRV-ATTR:kernel_id'];
        },
        set: function (oSEXTSRVATTRKernelId) {
            this['OS-EXT-SRV-ATTR:kernel_id'] = oSEXTSRVATTRKernelId;
        },
        enumerable: false,
        configurable: true
    });
    ServerDetail.prototype.withOSEXTSRVATTRRamdiskId = function (oSEXTSRVATTRRamdiskId) {
        this['OS-EXT-SRV-ATTR:ramdisk_id'] = oSEXTSRVATTRRamdiskId;
        return this;
    };
    Object.defineProperty(ServerDetail.prototype, "oSEXTSRVATTRRamdiskId", {
        get: function () {
            return this['OS-EXT-SRV-ATTR:ramdisk_id'];
        },
        set: function (oSEXTSRVATTRRamdiskId) {
            this['OS-EXT-SRV-ATTR:ramdisk_id'] = oSEXTSRVATTRRamdiskId;
        },
        enumerable: false,
        configurable: true
    });
    ServerDetail.prototype.withOSEXTSRVATTRRootDeviceName = function (oSEXTSRVATTRRootDeviceName) {
        this['OS-EXT-SRV-ATTR:root_device_name'] = oSEXTSRVATTRRootDeviceName;
        return this;
    };
    Object.defineProperty(ServerDetail.prototype, "oSEXTSRVATTRRootDeviceName", {
        get: function () {
            return this['OS-EXT-SRV-ATTR:root_device_name'];
        },
        set: function (oSEXTSRVATTRRootDeviceName) {
            this['OS-EXT-SRV-ATTR:root_device_name'] = oSEXTSRVATTRRootDeviceName;
        },
        enumerable: false,
        configurable: true
    });
    ServerDetail.prototype.withOSEXTSRVATTRUserData = function (oSEXTSRVATTRUserData) {
        this['OS-EXT-SRV-ATTR:user_data'] = oSEXTSRVATTRUserData;
        return this;
    };
    Object.defineProperty(ServerDetail.prototype, "oSEXTSRVATTRUserData", {
        get: function () {
            return this['OS-EXT-SRV-ATTR:user_data'];
        },
        set: function (oSEXTSRVATTRUserData) {
            this['OS-EXT-SRV-ATTR:user_data'] = oSEXTSRVATTRUserData;
        },
        enumerable: false,
        configurable: true
    });
    ServerDetail.prototype.withLocked = function (locked) {
        this['locked'] = locked;
        return this;
    };
    ServerDetail.prototype.withTags = function (tags) {
        this['tags'] = tags;
        return this;
    };
    ServerDetail.prototype.withOsSchedulerHints = function (osSchedulerHints) {
        this['os:scheduler_hints'] = osSchedulerHints;
        return this;
    };
    Object.defineProperty(ServerDetail.prototype, "osSchedulerHints", {
        get: function () {
            return this['os:scheduler_hints'];
        },
        set: function (osSchedulerHints) {
            this['os:scheduler_hints'] = osSchedulerHints;
        },
        enumerable: false,
        configurable: true
    });
    ServerDetail.prototype.withEnterpriseProjectId = function (enterpriseProjectId) {
        this['enterprise_project_id'] = enterpriseProjectId;
        return this;
    };
    Object.defineProperty(ServerDetail.prototype, "enterpriseProjectId", {
        get: function () {
            return this['enterprise_project_id'];
        },
        set: function (enterpriseProjectId) {
            this['enterprise_project_id'] = enterpriseProjectId;
        },
        enumerable: false,
        configurable: true
    });
    ServerDetail.prototype.withSysTags = function (sysTags) {
        this['sys_tags'] = sysTags;
        return this;
    };
    Object.defineProperty(ServerDetail.prototype, "sysTags", {
        get: function () {
            return this['sys_tags'];
        },
        set: function (sysTags) {
            this['sys_tags'] = sysTags;
        },
        enumerable: false,
        configurable: true
    });
    ServerDetail.prototype.withCpuOptions = function (cpuOptions) {
        this['cpu_options'] = cpuOptions;
        return this;
    };
    Object.defineProperty(ServerDetail.prototype, "cpuOptions", {
        get: function () {
            return this['cpu_options'];
        },
        set: function (cpuOptions) {
            this['cpu_options'] = cpuOptions;
        },
        enumerable: false,
        configurable: true
    });
    ServerDetail.prototype.withHypervisor = function (hypervisor) {
        this['hypervisor'] = hypervisor;
        return this;
    };
    return ServerDetail;
}());
exports.ServerDetail = ServerDetail;
