// @huaweicloud/huaweicloud-sdk-ecs/v2/model/ServerBlockDevice.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServerBlockDevice = void 0;
var ServerBlockDevice = /** @class */ (function () {
    function ServerBlockDevice() {
    }
    ServerBlockDevice.prototype.withBootIndex = function (bootIndex) {
        this['bootIndex'] = bootIndex;
        return this;
    };
    ServerBlockDevice.prototype.withPciAddress = function (pciAddress) {
        this['pciAddress'] = pciAddress;
        return this;
    };
    ServerBlockDevice.prototype.withVolumeId = function (volumeId) {
        this['volumeId'] = volumeId;
        return this;
    };
    ServerBlockDevice.prototype.withDevice = function (device) {
        this['device'] = device;
        return this;
    };
    ServerBlockDevice.prototype.withServerId = function (serverId) {
        this['serverId'] = serverId;
        return this;
    };
    ServerBlockDevice.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    ServerBlockDevice.prototype.withSize = function (size) {
        this['size'] = size;
        return this;
    };
    ServerBlockDevice.prototype.withBus = function (bus) {
        this['bus'] = bus;
        return this;
    };
    return ServerBlockDevice;
}());
exports.ServerBlockDevice = ServerBlockDevice;
