// @huaweicloud/huaweicloud-sdk-ecs/v2/model/ServerExtendVolumeAttachment.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServerExtendVolumeAttachment = void 0;
var ServerExtendVolumeAttachment = /** @class */ (function () {
    function ServerExtendVolumeAttachment(id, deleteOnTermination, device) {
        this['id'] = id;
        this['delete_on_termination'] = deleteOnTermination;
        this['device'] = device;
    }
    ServerExtendVolumeAttachment.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    ServerExtendVolumeAttachment.prototype.withDeleteOnTermination = function (deleteOnTermination) {
        this['delete_on_termination'] = deleteOnTermination;
        return this;
    };
    Object.defineProperty(ServerExtendVolumeAttachment.prototype, "deleteOnTermination", {
        get: function () {
            return this['delete_on_termination'];
        },
        set: function (deleteOnTermination) {
            this['delete_on_termination'] = deleteOnTermination;
        },
        enumerable: false,
        configurable: true
    });
    ServerExtendVolumeAttachment.prototype.withBootIndex = function (bootIndex) {
        this['bootIndex'] = bootIndex;
        return this;
    };
    ServerExtendVolumeAttachment.prototype.withDevice = function (device) {
        this['device'] = device;
        return this;
    };
    return ServerExtendVolumeAttachment;
}());
exports.ServerExtendVolumeAttachment = ServerExtendVolumeAttachment;
