// @huaweicloud/huaweicloud-sdk-ecs/v2/model/ServerAttachableQuantity.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServerAttachableQuantity = void 0;
var ServerAttachableQuantity = /** @class */ (function () {
    function ServerAttachableQuantity(freeScsi, freeBlk, freeDisk, freeNic) {
        this['free_scsi'] = freeScsi;
        this['free_blk'] = freeBlk;
        this['free_disk'] = freeDisk;
        this['free_nic'] = freeNic;
    }
    ServerAttachableQuantity.prototype.withFreeScsi = function (freeScsi) {
        this['free_scsi'] = freeScsi;
        return this;
    };
    Object.defineProperty(ServerAttachableQuantity.prototype, "freeScsi", {
        get: function () {
            return this['free_scsi'];
        },
        set: function (freeScsi) {
            this['free_scsi'] = freeScsi;
        },
        enumerable: false,
        configurable: true
    });
    ServerAttachableQuantity.prototype.withFreeBlk = function (freeBlk) {
        this['free_blk'] = freeBlk;
        return this;
    };
    Object.defineProperty(ServerAttachableQuantity.prototype, "freeBlk", {
        get: function () {
            return this['free_blk'];
        },
        set: function (freeBlk) {
            this['free_blk'] = freeBlk;
        },
        enumerable: false,
        configurable: true
    });
    ServerAttachableQuantity.prototype.withFreeDisk = function (freeDisk) {
        this['free_disk'] = freeDisk;
        return this;
    };
    Object.defineProperty(ServerAttachableQuantity.prototype, "freeDisk", {
        get: function () {
            return this['free_disk'];
        },
        set: function (freeDisk) {
            this['free_disk'] = freeDisk;
        },
        enumerable: false,
        configurable: true
    });
    ServerAttachableQuantity.prototype.withFreeNic = function (freeNic) {
        this['free_nic'] = freeNic;
        return this;
    };
    Object.defineProperty(ServerAttachableQuantity.prototype, "freeNic", {
        get: function () {
            return this['free_nic'];
        },
        set: function (freeNic) {
            this['free_nic'] = freeNic;
        },
        enumerable: false,
        configurable: true
    });
    return ServerAttachableQuantity;
}());
exports.ServerAttachableQuantity = ServerAttachableQuantity;
