// @huaweicloud/huaweicloud-sdk-ecs/v2/model/ServerAddress.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServerAddressOSEXTIPSTypeEnum = exports.ServerAddress = void 0;
var ServerAddress = /** @class */ (function () {
    function ServerAddress(version, addr) {
        this['version'] = version;
        this['addr'] = addr;
    }
    ServerAddress.prototype.withVersion = function (version) {
        this['version'] = version;
        return this;
    };
    ServerAddress.prototype.withAddr = function (addr) {
        this['addr'] = addr;
        return this;
    };
    ServerAddress.prototype.withOSEXTIPSType = function (oSEXTIPSType) {
        this['OS-EXT-IPS:type'] = oSEXTIPSType;
        return this;
    };
    Object.defineProperty(ServerAddress.prototype, "oSEXTIPSType", {
        get: function () {
            return this['OS-EXT-IPS:type'];
        },
        set: function (oSEXTIPSType) {
            this['OS-EXT-IPS:type'] = oSEXTIPSType;
        },
        enumerable: false,
        configurable: true
    });
    ServerAddress.prototype.withOSEXTIPSMACMacAddr = function (oSEXTIPSMACMacAddr) {
        this['OS-EXT-IPS-MAC:mac_addr'] = oSEXTIPSMACMacAddr;
        return this;
    };
    Object.defineProperty(ServerAddress.prototype, "oSEXTIPSMACMacAddr", {
        get: function () {
            return this['OS-EXT-IPS-MAC:mac_addr'];
        },
        set: function (oSEXTIPSMACMacAddr) {
            this['OS-EXT-IPS-MAC:mac_addr'] = oSEXTIPSMACMacAddr;
        },
        enumerable: false,
        configurable: true
    });
    ServerAddress.prototype.withOSEXTIPSPortId = function (oSEXTIPSPortId) {
        this['OS-EXT-IPS:port_id'] = oSEXTIPSPortId;
        return this;
    };
    Object.defineProperty(ServerAddress.prototype, "oSEXTIPSPortId", {
        get: function () {
            return this['OS-EXT-IPS:port_id'];
        },
        set: function (oSEXTIPSPortId) {
            this['OS-EXT-IPS:port_id'] = oSEXTIPSPortId;
        },
        enumerable: false,
        configurable: true
    });
    return ServerAddress;
}());
exports.ServerAddress = ServerAddress;
/**
    * @export
    * @enum {string}
    */
var ServerAddressOSEXTIPSTypeEnum;
(function (ServerAddressOSEXTIPSTypeEnum) {
    ServerAddressOSEXTIPSTypeEnum["FIXED"] = "fixed";
    ServerAddressOSEXTIPSTypeEnum["FLOATING"] = "floating";
})(ServerAddressOSEXTIPSTypeEnum = exports.ServerAddressOSEXTIPSTypeEnum || (exports.ServerAddressOSEXTIPSTypeEnum = {}));
