// @huaweicloud/huaweicloud-sdk-ecs/v2/model/ServerSchedulerHints.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServerSchedulerHints = void 0;
var ServerSchedulerHints = /** @class */ (function () {
    function ServerSchedulerHints() {
    }
    ServerSchedulerHints.prototype.withGroup = function (group) {
        this['group'] = group;
        return this;
    };
    ServerSchedulerHints.prototype.withTenancy = function (tenancy) {
        this['tenancy'] = tenancy;
        return this;
    };
    ServerSchedulerHints.prototype.withDedicatedHostId = function (dedicatedHostId) {
        this['dedicated_host_id'] = dedicatedHostId;
        return this;
    };
    Object.defineProperty(ServerSchedulerHints.prototype, "dedicatedHostId", {
        get: function () {
            return this['dedicated_host_id'];
        },
        set: function (dedicatedHostId) {
            this['dedicated_host_id'] = dedicatedHostId;
        },
        enumerable: false,
        configurable: true
    });
    return ServerSchedulerHints;
}());
exports.ServerSchedulerHints = ServerSchedulerHints;
