// @huaweicloud/huaweicloud-sdk-ecs/v2/model/ShowJobResponse.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShowJobResponseStatusEnum = exports.ShowJobResponse = void 0;
var SdkResponse_1 = require("@huaweicloud/huaweicloud-sdk-core/SdkResponse");
var ShowJobResponse = /** @class */ (function (_super) {
    __extends(ShowJobResponse, _super);
    function ShowJobResponse() {
        return _super.call(this) || this;
    }
    ShowJobResponse.prototype.withBeginTime = function (beginTime) {
        this['begin_time'] = beginTime;
        return this;
    };
    Object.defineProperty(ShowJobResponse.prototype, "beginTime", {
        get: function () {
            return this['begin_time'];
        },
        set: function (beginTime) {
            this['begin_time'] = beginTime;
        },
        enumerable: false,
        configurable: true
    });
    ShowJobResponse.prototype.withCode = function (code) {
        this['code'] = code;
        return this;
    };
    ShowJobResponse.prototype.withEndTime = function (endTime) {
        this['end_time'] = endTime;
        return this;
    };
    Object.defineProperty(ShowJobResponse.prototype, "endTime", {
        get: function () {
            return this['end_time'];
        },
        set: function (endTime) {
            this['end_time'] = endTime;
        },
        enumerable: false,
        configurable: true
    });
    ShowJobResponse.prototype.withEntities = function (entities) {
        this['entities'] = entities;
        return this;
    };
    ShowJobResponse.prototype.withErrorCode = function (errorCode) {
        this['error_code'] = errorCode;
        return this;
    };
    Object.defineProperty(ShowJobResponse.prototype, "errorCode", {
        get: function () {
            return this['error_code'];
        },
        set: function (errorCode) {
            this['error_code'] = errorCode;
        },
        enumerable: false,
        configurable: true
    });
    ShowJobResponse.prototype.withFailReason = function (failReason) {
        this['fail_reason'] = failReason;
        return this;
    };
    Object.defineProperty(ShowJobResponse.prototype, "failReason", {
        get: function () {
            return this['fail_reason'];
        },
        set: function (failReason) {
            this['fail_reason'] = failReason;
        },
        enumerable: false,
        configurable: true
    });
    ShowJobResponse.prototype.withJobId = function (jobId) {
        this['job_id'] = jobId;
        return this;
    };
    Object.defineProperty(ShowJobResponse.prototype, "jobId", {
        get: function () {
            return this['job_id'];
        },
        set: function (jobId) {
            this['job_id'] = jobId;
        },
        enumerable: false,
        configurable: true
    });
    ShowJobResponse.prototype.withJobType = function (jobType) {
        this['job_type'] = jobType;
        return this;
    };
    Object.defineProperty(ShowJobResponse.prototype, "jobType", {
        get: function () {
            return this['job_type'];
        },
        set: function (jobType) {
            this['job_type'] = jobType;
        },
        enumerable: false,
        configurable: true
    });
    ShowJobResponse.prototype.withMessage = function (message) {
        this['message'] = message;
        return this;
    };
    ShowJobResponse.prototype.withStatus = function (status) {
        this['status'] = status;
        return this;
    };
    return ShowJobResponse;
}(SdkResponse_1.SdkResponse));
exports.ShowJobResponse = ShowJobResponse;
/**
    * @export
    * @enum {string}
    */
var ShowJobResponseStatusEnum;
(function (ShowJobResponseStatusEnum) {
    ShowJobResponseStatusEnum["SUCCESS"] = "SUCCESS";
    ShowJobResponseStatusEnum["RUNNING"] = "RUNNING";
    ShowJobResponseStatusEnum["FAIL"] = "FAIL";
    ShowJobResponseStatusEnum["INIT"] = "INIT";
})(ShowJobResponseStatusEnum = exports.ShowJobResponseStatusEnum || (exports.ShowJobResponseStatusEnum = {}));
