// @huaweicloud/huaweicloud-sdk-ecs/v2/model/ShowServerGroupResult.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShowServerGroupResult = void 0;
var ShowServerGroupResult = /** @class */ (function () {
    function ShowServerGroupResult(id, members, metadata, name, policies) {
        this['id'] = id;
        this['members'] = members;
        this['metadata'] = metadata;
        this['name'] = name;
        this['policies'] = policies;
    }
    ShowServerGroupResult.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    ShowServerGroupResult.prototype.withMembers = function (members) {
        this['members'] = members;
        return this;
    };
    ShowServerGroupResult.prototype.withMetadata = function (metadata) {
        this['metadata'] = metadata;
        return this;
    };
    ShowServerGroupResult.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    ShowServerGroupResult.prototype.withPolicies = function (policies) {
        this['policies'] = policies;
        return this;
    };
    return ShowServerGroupResult;
}());
exports.ShowServerGroupResult = ShowServerGroupResult;
