// @huaweicloud/huaweicloud-sdk-ecs/v2/model/SubJobEntities.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubJobEntities = void 0;
var SubJobEntities = /** @class */ (function () {
    function SubJobEntities() {
    }
    SubJobEntities.prototype.withServerId = function (serverId) {
        this['server_id'] = serverId;
        return this;
    };
    Object.defineProperty(SubJobEntities.prototype, "serverId", {
        get: function () {
            return this['server_id'];
        },
        set: function (serverId) {
            this['server_id'] = serverId;
        },
        enumerable: false,
        configurable: true
    });
    SubJobEntities.prototype.withNicId = function (nicId) {
        this['nic_id'] = nicId;
        return this;
    };
    Object.defineProperty(SubJobEntities.prototype, "nicId", {
        get: function () {
            return this['nic_id'];
        },
        set: function (nicId) {
            this['nic_id'] = nicId;
        },
        enumerable: false,
        configurable: true
    });
    SubJobEntities.prototype.withErrorcodeMessage = function (errorcodeMessage) {
        this['errorcode_message'] = errorcodeMessage;
        return this;
    };
    Object.defineProperty(SubJobEntities.prototype, "errorcodeMessage", {
        get: function () {
            return this['errorcode_message'];
        },
        set: function (errorcodeMessage) {
            this['errorcode_message'] = errorcodeMessage;
        },
        enumerable: false,
        configurable: true
    });
    return SubJobEntities;
}());
exports.SubJobEntities = SubJobEntities;
