// @huaweicloud/huaweicloud-sdk-ecs/v2/model/SubJob.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubJobStatusEnum = exports.SubJob = void 0;
var SubJob = /** @class */ (function () {
    function SubJob() {
    }
    SubJob.prototype.withStatus = function (status) {
        this['status'] = status;
        return this;
    };
    SubJob.prototype.withEntities = function (entities) {
        this['entities'] = entities;
        return this;
    };
    SubJob.prototype.withJobId = function (jobId) {
        this['job_id'] = jobId;
        return this;
    };
    Object.defineProperty(SubJob.prototype, "jobId", {
        get: function () {
            return this['job_id'];
        },
        set: function (jobId) {
            this['job_id'] = jobId;
        },
        enumerable: false,
        configurable: true
    });
    SubJob.prototype.withJobType = function (jobType) {
        this['job_type'] = jobType;
        return this;
    };
    Object.defineProperty(SubJob.prototype, "jobType", {
        get: function () {
            return this['job_type'];
        },
        set: function (jobType) {
            this['job_type'] = jobType;
        },
        enumerable: false,
        configurable: true
    });
    SubJob.prototype.withBeginTime = function (beginTime) {
        this['begin_time'] = beginTime;
        return this;
    };
    Object.defineProperty(SubJob.prototype, "beginTime", {
        get: function () {
            return this['begin_time'];
        },
        set: function (beginTime) {
            this['begin_time'] = beginTime;
        },
        enumerable: false,
        configurable: true
    });
    SubJob.prototype.withEndTime = function (endTime) {
        this['end_time'] = endTime;
        return this;
    };
    Object.defineProperty(SubJob.prototype, "endTime", {
        get: function () {
            return this['end_time'];
        },
        set: function (endTime) {
            this['end_time'] = endTime;
        },
        enumerable: false,
        configurable: true
    });
    SubJob.prototype.withErrorCode = function (errorCode) {
        this['error_code'] = errorCode;
        return this;
    };
    Object.defineProperty(SubJob.prototype, "errorCode", {
        get: function () {
            return this['error_code'];
        },
        set: function (errorCode) {
            this['error_code'] = errorCode;
        },
        enumerable: false,
        configurable: true
    });
    SubJob.prototype.withFailReason = function (failReason) {
        this['fail_reason'] = failReason;
        return this;
    };
    Object.defineProperty(SubJob.prototype, "failReason", {
        get: function () {
            return this['fail_reason'];
        },
        set: function (failReason) {
            this['fail_reason'] = failReason;
        },
        enumerable: false,
        configurable: true
    });
    return SubJob;
}());
exports.SubJob = SubJob;
/**
    * @export
    * @enum {string}
    */
var SubJobStatusEnum;
(function (SubJobStatusEnum) {
    SubJobStatusEnum["SUCCESS"] = "SUCCESS";
    SubJobStatusEnum["RUNNING"] = "RUNNING";
    SubJobStatusEnum["FAIL"] = "FAIL";
    SubJobStatusEnum["INIT"] = "INIT";
})(SubJobStatusEnum = exports.SubJobStatusEnum || (exports.SubJobStatusEnum = {}));
